// @sec-ant/zxing-wasm/dist/reader/index.js
import { g as ai, r as oi, d as V, a as si } from "../core-469ea631.js";
import { Z as pi, b as hi, c as _i, p as gi, s as yi } from "../core-469ea631.js";
var Ve = (() => {
  var x = import.meta.url;
  return function(N = {}) {
    var u = typeof N < "u" ? N : {}, G, ce;
    u.ready = new Promise((e, r) => {
      G = e, ce = r;
    });
    var Ne = Object.assign({}, u), Be = "./this.program", Le = typeof window == "object", le = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var W = "";
    function Dr(e) {
      return u.locateFile ? u.locateFile(e, W) : W + e;
    }
    var Fe;
    (Le || le) && (le ? W = self.location.href : typeof document < "u" && document.currentScript && (W = document.currentScript.src), x && (W = x), W.indexOf("blob:") !== 0 ? W = W.substr(0, W.replace(/[?#].*/, "").lastIndexOf("/") + 1) : W = "", le && (Fe = (e) => {
      var r = new XMLHttpRequest();
      return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), u.print || console.log.bind(console);
    var ee = u.printErr || console.error.bind(console);
    Object.assign(u, Ne), Ne = null, u.arguments && u.arguments, u.thisProgram && (Be = u.thisProgram), u.quit && u.quit;
    var re;
    u.wasmBinary && (re = u.wasmBinary), u.noExitRuntime, typeof WebAssembly != "object" && J("no native wasm support detected");
    var de, ze = !1;
    function Or(e, r) {
      e || J(r);
    }
    var U, k, q, ve, F, T, Xe, Ze;
    function Ge() {
      var e = de.buffer;
      u.HEAP8 = U = new Int8Array(e), u.HEAP16 = q = new Int16Array(e), u.HEAP32 = F = new Int32Array(e), u.HEAPU8 = k = new Uint8Array(e), u.HEAPU16 = ve = new Uint16Array(e), u.HEAPU32 = T = new Uint32Array(e), u.HEAPF32 = Xe = new Float32Array(e), u.HEAPF64 = Ze = new Float64Array(e);
    }
    var qe, Je = [], Ke = [], Qe = [];
    function Wr() {
      if (u.preRun)
        for (typeof u.preRun == "function" && (u.preRun = [u.preRun]); u.preRun.length; )
          jr(u.preRun.shift());
      Ee(Je);
    }
    function Ur() {
      Ee(Ke);
    }
    function Hr() {
      if (u.postRun)
        for (typeof u.postRun == "function" && (u.postRun = [u.postRun]); u.postRun.length; )
          Ir(u.postRun.shift());
      Ee(Qe);
    }
    function jr(e) {
      Je.unshift(e);
    }
    function Mr(e) {
      Ke.unshift(e);
    }
    function Ir(e) {
      Qe.unshift(e);
    }
    var B = 0, te = null;
    function xr(e) {
      B++, u.monitorRunDependencies && u.monitorRunDependencies(B);
    }
    function Yr(e) {
      if (B--, u.monitorRunDependencies && u.monitorRunDependencies(B), B == 0 && te) {
        var r = te;
        te = null, r();
      }
    }
    function J(e) {
      u.onAbort && u.onAbort(e), e = "Aborted(" + e + ")", ee(e), ze = !0, e += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(e);
      throw ce(r), r;
    }
    var Vr = "data:application/octet-stream;base64,";
    function er(e) {
      return e.startsWith(Vr);
    }
    var L;
    u.locateFile ? (L = "zxing_reader.wasm", er(L) || (L = Dr(L))) : L = new URL("/reader/zxing_reader.wasm", self.location).href;
    function rr(e) {
      try {
        if (e == L && re)
          return new Uint8Array(re);
        if (Fe)
          return Fe(e);
        throw "both async and sync fetching of the wasm failed";
      } catch (r) {
        J(r);
      }
    }
    function Nr(e) {
      return !re && (Le || le) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return r.arrayBuffer();
      }).catch(() => rr(e)) : Promise.resolve().then(() => rr(e));
    }
    function tr(e, r, t) {
      return Nr(e).then((n) => WebAssembly.instantiate(n, r)).then((n) => n).then(t, (n) => {
        ee("failed to asynchronously prepare wasm: " + n), J(n);
      });
    }
    function Br(e, r, t, n) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !er(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((i) => {
        var a = WebAssembly.instantiateStreaming(i, t);
        return a.then(n, function(s) {
          return ee("wasm streaming compile failed: " + s), ee("falling back to ArrayBuffer instantiation"), tr(r, t, n);
        });
      }) : tr(r, t, n);
    }
    function Lr() {
      var e = { a: An };
      function r(n, i) {
        var a = n.exports;
        return u.asm = a, de = u.asm.oa, Ge(), qe = u.asm.sa, Mr(u.asm.pa), Yr(), a;
      }
      xr();
      function t(n) {
        r(n.instance);
      }
      if (u.instantiateWasm)
        try {
          return u.instantiateWasm(e, r);
        } catch (n) {
          ee("Module.instantiateWasm callback failed with error: " + n), ce(n);
        }
      return Br(re, L, e, t).catch(ce), {};
    }
    function Ee(e) {
      for (; e.length > 0; )
        e.shift()(u);
    }
    var pe = [], he = 0;
    function zr(e) {
      var r = new _e(e);
      return r.get_caught() || (r.set_caught(!0), he--), r.set_rethrown(!1), pe.push(r), Ar(r.excPtr), r.get_exception_ptr();
    }
    var H = 0;
    function Xr() {
      _(0);
      var e = pe.pop();
      Tr(e.excPtr), H = 0;
    }
    function _e(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(r) {
        T[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return T[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        T[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return T[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, U[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return U[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, U[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return U[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }, this.set_adjusted_ptr = function(r) {
        T[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return T[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Er(this.get_type());
        if (r)
          return T[this.excPtr >> 2];
        var t = this.get_adjusted_ptr();
        return t !== 0 ? t : this.excPtr;
      };
    }
    function Zr(e) {
      throw H || (H = e), H;
    }
    function Se() {
      var e = H;
      if (!e)
        return fe(0), 0;
      var r = new _e(e);
      r.set_adjusted_ptr(e);
      var t = r.get_type();
      if (!t)
        return fe(0), e;
      for (var n = 0; n < arguments.length; n++) {
        var i = arguments[n];
        if (i === 0 || i === t)
          break;
        var a = r.ptr + 16;
        if (Fr(i, t, a))
          return fe(i), e;
      }
      return fe(t), e;
    }
    var Gr = Se, qr = Se, Jr = Se;
    function Kr(e) {
      var r = new _e(e).get_exception_ptr();
      return r;
    }
    function Qr() {
      var e = pe.pop();
      e || J("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (pe.push(e), e.set_rethrown(!0), e.set_caught(!1), he++), H = r, H;
    }
    function et(e, r, t) {
      var n = new _e(e);
      throw n.init(r, t), H = e, he++, H;
    }
    function rt() {
      return he;
    }
    var ge = {};
    function nr(e) {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    }
    function ne(e) {
      return this.fromWireType(F[e >> 2]);
    }
    var K = {}, z = {}, ye = {}, tt = 48, nt = 57;
    function ir(e) {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= tt && r <= nt ? `_${e}` : e;
    }
    function ar(e, r) {
      return e = ir(e), { [e]: function() {
        return r.apply(this, arguments);
      } }[e];
    }
    function Re(e, r) {
      var t = ar(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }
    var or = void 0;
    function me(e) {
      throw new or(e);
    }
    function X(e, r, t) {
      e.forEach(function(o) {
        ye[o] = r;
      });
      function n(o) {
        var f = t(o);
        f.length !== e.length && me("Mismatched type converter count");
        for (var c = 0; c < e.length; ++c)
          j(e[c], f[c]);
      }
      var i = new Array(r.length), a = [], s = 0;
      r.forEach((o, f) => {
        z.hasOwnProperty(o) ? i[f] = z[o] : (a.push(o), K.hasOwnProperty(o) || (K[o] = []), K[o].push(() => {
          i[f] = z[o], ++s, s === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }
    function it(e) {
      var r = ge[e];
      delete ge[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((s) => s.getterReturnType).concat(i.map((s) => s.setterArgumentType));
      X([e], a, (s) => {
        var o = {};
        return i.forEach((f, c) => {
          var d = f.fieldName, p = s[c], h = f.getter, m = f.getterContext, $ = s[c + i.length], E = f.setter, C = f.setterContext;
          o[d] = { read: (S) => p.fromWireType(h(m, S)), write: (S, l) => {
            var v = [];
            E(C, S, $.toWireType(v, l)), nr(v);
          } };
        }), [{ name: r.name, fromWireType: function(f) {
          var c = {};
          for (var d in o)
            c[d] = o[d].read(f);
          return n(f), c;
        }, toWireType: function(f, c) {
          for (var d in o)
            if (!(d in c))
              throw new TypeError(`Missing field: "${d}"`);
          var p = t();
          for (d in o)
            o[d].write(p, c[d]);
          return f !== null && f.push(n, p), p;
        }, argPackAdvance: 8, readValueFromPointer: ne, destructorFunction: n }];
      });
    }
    function at(e, r, t, n, i) {
    }
    function ke(e) {
      switch (e) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${e}`);
      }
    }
    function ot() {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      sr = e;
    }
    var sr = void 0;
    function R(e) {
      for (var r = "", t = e; k[t]; )
        r += sr[k[t++]];
      return r;
    }
    var Q = void 0;
    function w(e) {
      throw new Q(e);
    }
    function j(e, r, t = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      var n = r.name;
      if (e || w(`type "${n}" must have a positive integer typeid pointer`), z.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        w(`Cannot register type '${n}' twice`);
      }
      if (z[e] = r, delete ye[e], K.hasOwnProperty(e)) {
        var i = K[e];
        delete K[e], i.forEach((a) => a());
      }
    }
    function st(e, r, t, n, i) {
      var a = ke(t);
      r = R(r), j(e, { name: r, fromWireType: function(s) {
        return !!s;
      }, toWireType: function(s, o) {
        return o ? n : i;
      }, argPackAdvance: 8, readValueFromPointer: function(s) {
        var o;
        if (t === 1)
          o = U;
        else if (t === 2)
          o = q;
        else if (t === 4)
          o = F;
        else
          throw new TypeError("Unknown boolean type size: " + r);
        return this.fromWireType(o[s >> a]);
      }, destructorFunction: null });
    }
    function ut(e) {
      if (!(this instanceof Y) || !(e instanceof Y))
        return !1;
      for (var r = this.$$.ptrType.registeredClass, t = this.$$.ptr, n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
        t = r.upcast(t), r = r.baseClass;
      for (; n.baseClass; )
        i = n.upcast(i), n = n.baseClass;
      return r === n && t === i;
    }
    function ft(e) {
      return { count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType };
    }
    function De(e) {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      w(r(e) + " instance already deleted");
    }
    var Oe = !1;
    function ur(e) {
    }
    function ct(e) {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }
    function fr(e) {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && ct(e);
    }
    function cr(e, r, t) {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = cr(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }
    var lr = {};
    function lt() {
      return Object.keys(oe).length;
    }
    function dt() {
      var e = [];
      for (var r in oe)
        oe.hasOwnProperty(r) && e.push(oe[r]);
      return e;
    }
    var ie = [];
    function We() {
      for (; ie.length; ) {
        var e = ie.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }
    var ae = void 0;
    function vt(e) {
      ae = e, ie.length && ae && ae(We);
    }
    function pt() {
      u.getInheritedInstanceCount = lt, u.getLiveInheritedInstances = dt, u.flushPendingDeletes = We, u.setDelayFunction = vt;
    }
    var oe = {};
    function ht(e, r) {
      for (r === void 0 && w("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }
    function _t(e, r) {
      return r = ht(e, r), oe[r];
    }
    function we(e, r) {
      (!r.ptrType || !r.ptr) && me("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && me("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, se(Object.create(e, { $$: { value: r } }));
    }
    function gt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = _t(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? we(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: e }) : we(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var a = this.registeredClass.getActualType(r), s = lr[a];
      if (!s)
        return i.call(this);
      var o;
      this.isConst ? o = s.constPointerType : o = s.pointerType;
      var f = cr(r, this.registeredClass, o.registeredClass);
      return f === null ? i.call(this) : this.isSmartPointer ? we(o.registeredClass.instancePrototype, { ptrType: o, ptr: f, smartPtrType: this, smartPtr: e }) : we(o.registeredClass.instancePrototype, { ptrType: o, ptr: f });
    }
    function se(e) {
      return typeof FinalizationRegistry > "u" ? (se = (r) => r, e) : (Oe = new FinalizationRegistry((r) => {
        fr(r.$$);
      }), se = (r) => {
        var t = r.$$, n = !!t.smartPtr;
        if (n) {
          var i = { $$: t };
          Oe.register(r, i, r);
        }
        return r;
      }, ur = (r) => Oe.unregister(r), se(e));
    }
    function yt() {
      if (this.$$.ptr || De(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var e = se(Object.create(Object.getPrototypeOf(this), { $$: { value: ft(this.$$) } }));
      return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
    }
    function mt() {
      this.$$.ptr || De(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), ur(this), fr(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function wt() {
      return !this.$$.ptr;
    }
    function bt() {
      return this.$$.ptr || De(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && w("Object already scheduled for deletion"), ie.push(this), ie.length === 1 && ae && ae(We), this.$$.deleteScheduled = !0, this;
    }
    function $t() {
      Y.prototype.isAliasOf = ut, Y.prototype.clone = yt, Y.prototype.delete = mt, Y.prototype.isDeleted = wt, Y.prototype.deleteLater = bt;
    }
    function Y() {
    }
    function dr(e, r, t) {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          return e[r].overloadTable.hasOwnProperty(arguments.length) || w(`Function '${t}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[arguments.length].apply(this, arguments);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }
    function vr(e, r, t) {
      u.hasOwnProperty(e) ? ((t === void 0 || u[e].overloadTable !== void 0 && u[e].overloadTable[t] !== void 0) && w(`Cannot register public name '${e}' twice`), dr(u, e, e), u.hasOwnProperty(t) && w(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), u[e].overloadTable[t] = r) : (u[e] = r, t !== void 0 && (u[e].numArguments = t));
    }
    function Ct(e, r, t, n, i, a, s, o) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = s, this.downcast = o, this.pureVirtualFunctions = [];
    }
    function Ue(e, r, t) {
      for (; r !== t; )
        r.upcast || w(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    }
    function Pt(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ue(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Tt(e, r) {
      var t;
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = Ue(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && w("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : w(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, Ce.toHandle(function() {
                i.delete();
              })), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            w("Unsupporting sharing policy");
        }
      return t;
    }
    function At(e, r) {
      if (r === null)
        return this.isReference && w(`null is not a valid ${this.name}`), 0;
      r.$$ || w(`Cannot pass "${Me(r)}" as a ${this.name}`), r.$$.ptr || w(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && w(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Ue(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Ft(e) {
      return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
    }
    function Et(e) {
      this.rawDestructor && this.rawDestructor(e);
    }
    function St(e) {
      e !== null && e.delete();
    }
    function Rt() {
      M.prototype.getPointee = Ft, M.prototype.destructor = Et, M.prototype.argPackAdvance = 8, M.prototype.readValueFromPointer = ne, M.prototype.deleteObject = St, M.prototype.fromWireType = gt;
    }
    function M(e, r, t, n, i, a, s, o, f, c, d) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = s, this.rawGetPointee = o, this.rawConstructor = f, this.rawShare = c, this.rawDestructor = d, !i && r.baseClass === void 0 ? n ? (this.toWireType = Pt, this.destructorFunction = null) : (this.toWireType = At, this.destructorFunction = null) : this.toWireType = Tt;
    }
    function pr(e, r, t) {
      u.hasOwnProperty(e) || me("Replacing nonexistant public symbol"), u[e].overloadTable !== void 0 && t !== void 0 ? u[e].overloadTable[t] = r : (u[e] = r, u[e].argCount = t);
    }
    function kt(e, r, t) {
      var n = u["dynCall_" + e];
      return t && t.length ? n.apply(null, [r].concat(t)) : n.call(null, r);
    }
    var be = [];
    function b(e) {
      var r = be[e];
      return r || (e >= be.length && (be.length = e + 1), be[e] = r = qe.get(e)), r;
    }
    function Dt(e, r, t) {
      if (e.includes("j"))
        return kt(e, r, t);
      var n = b(r).apply(null, t);
      return n;
    }
    function Ot(e, r) {
      var t = [];
      return function() {
        return t.length = 0, Object.assign(t, arguments), Dt(e, r, t);
      };
    }
    function O(e, r) {
      e = R(e);
      function t() {
        return e.includes("j") ? Ot(e, r) : b(r);
      }
      var n = t();
      return typeof n != "function" && w(`unknown function pointer with signature ${e}: ${r}`), n;
    }
    var hr = void 0;
    function _r(e) {
      var r = Pr(e), t = R(r);
      return I(r), t;
    }
    function $e(e, r) {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !z[a]) {
          if (ye[a]) {
            ye[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new hr(`${e}: ` + t.map(_r).join([", "]));
    }
    function Wt(e, r, t, n, i, a, s, o, f, c, d, p, h) {
      d = R(d), a = O(i, a), o && (o = O(s, o)), c && (c = O(f, c)), h = O(p, h);
      var m = ir(d);
      vr(m, function() {
        $e(`Cannot construct ${d} due to unbound types`, [n]);
      }), X([e, r, t], n ? [n] : [], function($) {
        $ = $[0];
        var E, C;
        n ? (E = $.registeredClass, C = E.instancePrototype) : C = Y.prototype;
        var S = ar(m, function() {
          if (Object.getPrototypeOf(this) !== l)
            throw new Q("Use 'new' to construct " + d);
          if (v.constructor_body === void 0)
            throw new Q(d + " has no accessible constructor");
          var Ae = v.constructor_body[arguments.length];
          if (Ae === void 0)
            throw new Q(`Tried to invoke ctor of ${d} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(v.constructor_body).toString()}) parameters instead!`);
          return Ae.apply(this, arguments);
        }), l = Object.create(C, { constructor: { value: S } });
        S.prototype = l;
        var v = new Ct(d, S, l, h, E, a, o, c);
        v.baseClass && (v.baseClass.__derivedClasses === void 0 && (v.baseClass.__derivedClasses = []), v.baseClass.__derivedClasses.push(v));
        var P = new M(d, v, !0, !1, !1), A = new M(d + "*", v, !1, !1, !1), Z = new M(d + " const*", v, !1, !0, !1);
        return lr[e] = { pointerType: A, constPointerType: Z }, pr(m, S), [P, A, Z];
      });
    }
    function He(e, r) {
      for (var t = [], n = 0; n < e; n++)
        t.push(T[r + n * 4 >> 2]);
      return t;
    }
    function je(e, r, t, n, i, a) {
      var s = r.length;
      s < 2 && w("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var o = r[1] !== null && t !== null, f = !1, c = 1; c < r.length; ++c)
        if (r[c] !== null && r[c].destructorFunction === void 0) {
          f = !0;
          break;
        }
      var d = r[0].name !== "void", p = s - 2, h = new Array(p), m = [], $ = [];
      return function() {
        arguments.length !== p && w(`function ${e} called with ${arguments.length} arguments, expected ${p} args!`), $.length = 0;
        var E;
        m.length = o ? 2 : 1, m[0] = i, o && (E = r[1].toWireType($, this), m[1] = E);
        for (var C = 0; C < p; ++C)
          h[C] = r[C + 2].toWireType($, arguments[C]), m.push(h[C]);
        var S = n.apply(null, m);
        function l(v) {
          if (f)
            nr($);
          else
            for (var P = o ? 1 : 2; P < r.length; P++) {
              var A = P === 1 ? E : h[P - 2];
              r[P].destructorFunction !== null && r[P].destructorFunction(A);
            }
          if (d)
            return r[0].fromWireType(v);
        }
        return l(S);
      };
    }
    function Ut(e, r, t, n, i, a) {
      Or(r > 0);
      var s = He(r, t);
      i = O(n, i), X([], [e], function(o) {
        o = o[0];
        var f = `constructor ${o.name}`;
        if (o.registeredClass.constructor_body === void 0 && (o.registeredClass.constructor_body = []), o.registeredClass.constructor_body[r - 1] !== void 0)
          throw new Q(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${o.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return o.registeredClass.constructor_body[r - 1] = () => {
          $e(`Cannot construct ${o.name} due to unbound types`, s);
        }, X([], s, function(c) {
          return c.splice(1, 0, null), o.registeredClass.constructor_body[r - 1] = je(f, c, null, i, a), [];
        }), [];
      });
    }
    function Ht(e, r, t, n, i, a, s, o, f) {
      var c = He(t, n);
      r = R(r), a = O(i, a), X([], [e], function(d) {
        d = d[0];
        var p = `${d.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), o && d.registeredClass.pureVirtualFunctions.push(r);
        function h() {
          $e(`Cannot call ${p} due to unbound types`, c);
        }
        var m = d.registeredClass.instancePrototype, $ = m[r];
        return $ === void 0 || $.overloadTable === void 0 && $.className !== d.name && $.argCount === t - 2 ? (h.argCount = t - 2, h.className = d.name, m[r] = h) : (dr(m, r, p), m[r].overloadTable[t - 2] = h), X([], c, function(E) {
          var C = je(p, E, d, a, s);
          return m[r].overloadTable === void 0 ? (C.argCount = t - 2, m[r] = C) : m[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    }
    function jt() {
      this.allocated = [void 0], this.freelist = [], this.get = function(e) {
        return this.allocated[e];
      }, this.has = function(e) {
        return this.allocated[e] !== void 0;
      }, this.allocate = function(e) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = e, r;
      }, this.free = function(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      };
    }
    var D = new jt();
    function gr(e) {
      e >= D.reserved && --D.get(e).refcount === 0 && D.free(e);
    }
    function Mt() {
      for (var e = 0, r = D.reserved; r < D.allocated.length; ++r)
        D.allocated[r] !== void 0 && ++e;
      return e;
    }
    function It() {
      D.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), D.reserved = D.allocated.length, u.count_emval_handles = Mt;
    }
    var Ce = { toValue: (e) => (e || w("Cannot use deleted val. handle = " + e), D.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return D.allocate({ refcount: 1, value: e });
      }
    } };
    function xt(e, r) {
      r = R(r), j(e, { name: r, fromWireType: function(t) {
        var n = Ce.toValue(t);
        return gr(t), n;
      }, toWireType: function(t, n) {
        return Ce.toHandle(n);
      }, argPackAdvance: 8, readValueFromPointer: ne, destructorFunction: null });
    }
    function Me(e) {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }
    function Yt(e, r) {
      switch (r) {
        case 2:
          return function(t) {
            return this.fromWireType(Xe[t >> 2]);
          };
        case 3:
          return function(t) {
            return this.fromWireType(Ze[t >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + e);
      }
    }
    function Vt(e, r, t) {
      var n = ke(t);
      r = R(r), j(e, { name: r, fromWireType: function(i) {
        return i;
      }, toWireType: function(i, a) {
        return a;
      }, argPackAdvance: 8, readValueFromPointer: Yt(r, n), destructorFunction: null });
    }
    function Nt(e, r, t, n, i, a, s) {
      var o = He(r, t);
      e = R(e), i = O(n, i), vr(e, function() {
        $e(`Cannot call ${e} due to unbound types`, o);
      }, r - 1), X([], o, function(f) {
        var c = [f[0], null].concat(f.slice(1));
        return pr(e, je(e, c, null, i, a), r - 1), [];
      });
    }
    function Bt(e, r, t) {
      switch (r) {
        case 0:
          return t ? function(i) {
            return U[i];
          } : function(i) {
            return k[i];
          };
        case 1:
          return t ? function(i) {
            return q[i >> 1];
          } : function(i) {
            return ve[i >> 1];
          };
        case 2:
          return t ? function(i) {
            return F[i >> 2];
          } : function(i) {
            return T[i >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + e);
      }
    }
    function Lt(e, r, t, n, i) {
      r = R(r);
      var a = ke(t), s = (p) => p;
      if (n === 0) {
        var o = 32 - 8 * t;
        s = (p) => p << o >>> o;
      }
      var f = r.includes("unsigned"), c = (p, h) => {
      }, d;
      f ? d = function(p, h) {
        return c(h, this.name), h >>> 0;
      } : d = function(p, h) {
        return c(h, this.name), h;
      }, j(e, { name: r, fromWireType: s, toWireType: d, argPackAdvance: 8, readValueFromPointer: Bt(r, a, n !== 0), destructorFunction: null });
    }
    function zt(e, r, t) {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(s) {
        s = s >> 2;
        var o = T, f = o[s], c = o[s + 1];
        return new i(o.buffer, c, f);
      }
      t = R(t), j(e, { name: t, fromWireType: a, argPackAdvance: 8, readValueFromPointer: a }, { ignoreDuplicateRegistrations: !0 });
    }
    function yr(e, r, t, n) {
      if (!(n > 0))
        return 0;
      for (var i = t, a = t + n - 1, s = 0; s < e.length; ++s) {
        var o = e.charCodeAt(s);
        if (o >= 55296 && o <= 57343) {
          var f = e.charCodeAt(++s);
          o = 65536 + ((o & 1023) << 10) | f & 1023;
        }
        if (o <= 127) {
          if (t >= a)
            break;
          r[t++] = o;
        } else if (o <= 2047) {
          if (t + 1 >= a)
            break;
          r[t++] = 192 | o >> 6, r[t++] = 128 | o & 63;
        } else if (o <= 65535) {
          if (t + 2 >= a)
            break;
          r[t++] = 224 | o >> 12, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        } else {
          if (t + 3 >= a)
            break;
          r[t++] = 240 | o >> 18, r[t++] = 128 | o >> 12 & 63, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        }
      }
      return r[t] = 0, t - i;
    }
    function Xt(e, r, t) {
      return yr(e, k, r, t);
    }
    function mr(e) {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }
    var wr = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0;
    function Zt(e, r, t) {
      for (var n = r + t, i = r; e[i] && !(i >= n); )
        ++i;
      if (i - r > 16 && e.buffer && wr)
        return wr.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var s = e[r++];
        if (!(s & 128)) {
          a += String.fromCharCode(s);
          continue;
        }
        var o = e[r++] & 63;
        if ((s & 224) == 192) {
          a += String.fromCharCode((s & 31) << 6 | o);
          continue;
        }
        var f = e[r++] & 63;
        if ((s & 240) == 224 ? s = (s & 15) << 12 | o << 6 | f : s = (s & 7) << 18 | o << 12 | f << 6 | e[r++] & 63, s < 65536)
          a += String.fromCharCode(s);
        else {
          var c = s - 65536;
          a += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        }
      }
      return a;
    }
    function Ie(e, r) {
      return e ? Zt(k, e, r) : "";
    }
    function Gt(e, r) {
      r = R(r);
      var t = r === "std::string";
      j(e, { name: r, fromWireType: function(n) {
        var i = T[n >> 2], a = n + 4, s;
        if (t)
          for (var o = a, f = 0; f <= i; ++f) {
            var c = a + f;
            if (f == i || k[c] == 0) {
              var d = c - o, p = Ie(o, d);
              s === void 0 ? s = p : (s += String.fromCharCode(0), s += p), o = c + 1;
            }
          }
        else {
          for (var h = new Array(i), f = 0; f < i; ++f)
            h[f] = String.fromCharCode(k[a + f]);
          s = h.join("");
        }
        return I(n), s;
      }, toWireType: function(n, i) {
        i instanceof ArrayBuffer && (i = new Uint8Array(i));
        var a, s = typeof i == "string";
        s || i instanceof Uint8Array || i instanceof Uint8ClampedArray || i instanceof Int8Array || w("Cannot pass non-string to std::string"), t && s ? a = mr(i) : a = i.length;
        var o = Ye(4 + a + 1), f = o + 4;
        if (T[o >> 2] = a, t && s)
          Xt(i, f, a + 1);
        else if (s)
          for (var c = 0; c < a; ++c) {
            var d = i.charCodeAt(c);
            d > 255 && (I(f), w("String has UTF-16 code units that do not fit in 8 bits")), k[f + c] = d;
          }
        else
          for (var c = 0; c < a; ++c)
            k[f + c] = i[c];
        return n !== null && n.push(I, o), o;
      }, argPackAdvance: 8, readValueFromPointer: ne, destructorFunction: function(n) {
        I(n);
      } });
    }
    var br = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0;
    function qt(e, r) {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && ve[n]; )
        ++n;
      if (t = n << 1, t - e > 32 && br)
        return br.decode(k.subarray(e, t));
      for (var a = "", s = 0; !(s >= r / 2); ++s) {
        var o = q[e + s * 2 >> 1];
        if (o == 0)
          break;
        a += String.fromCharCode(o);
      }
      return a;
    }
    function Jt(e, r, t) {
      if (t === void 0 && (t = 2147483647), t < 2)
        return 0;
      t -= 2;
      for (var n = r, i = t < e.length * 2 ? t / 2 : e.length, a = 0; a < i; ++a) {
        var s = e.charCodeAt(a);
        q[r >> 1] = s, r += 2;
      }
      return q[r >> 1] = 0, r - n;
    }
    function Kt(e) {
      return e.length * 2;
    }
    function Qt(e, r) {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = F[e + t * 4 >> 2];
        if (i == 0)
          break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }
    function en(e, r, t) {
      if (t === void 0 && (t = 2147483647), t < 4)
        return 0;
      for (var n = r, i = n + t - 4, a = 0; a < e.length; ++a) {
        var s = e.charCodeAt(a);
        if (s >= 55296 && s <= 57343) {
          var o = e.charCodeAt(++a);
          s = 65536 + ((s & 1023) << 10) | o & 1023;
        }
        if (F[r >> 2] = s, r += 4, r + 4 > i)
          break;
      }
      return F[r >> 2] = 0, r - n;
    }
    function rn(e) {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }
    function tn(e, r, t) {
      t = R(t);
      var n, i, a, s, o;
      r === 2 ? (n = qt, i = Jt, s = Kt, a = () => ve, o = 1) : r === 4 && (n = Qt, i = en, s = rn, a = () => T, o = 2), j(e, { name: t, fromWireType: function(f) {
        for (var c = T[f >> 2], d = a(), p, h = f + 4, m = 0; m <= c; ++m) {
          var $ = f + 4 + m * r;
          if (m == c || d[$ >> o] == 0) {
            var E = $ - h, C = n(h, E);
            p === void 0 ? p = C : (p += String.fromCharCode(0), p += C), h = $ + r;
          }
        }
        return I(f), p;
      }, toWireType: function(f, c) {
        typeof c != "string" && w(`Cannot pass non-string to C++ string type ${t}`);
        var d = s(c), p = Ye(4 + d + r);
        return T[p >> 2] = d >> o, i(c, p + 4, d + r), f !== null && f.push(I, p), p;
      }, argPackAdvance: 8, readValueFromPointer: ne, destructorFunction: function(f) {
        I(f);
      } });
    }
    function nn(e, r, t, n, i, a) {
      ge[e] = { name: R(r), rawConstructor: O(t, n), rawDestructor: O(i, a), fields: [] };
    }
    function an(e, r, t, n, i, a, s, o, f, c) {
      ge[e].fields.push({ fieldName: R(r), getterReturnType: t, getter: O(n, i), getterContext: a, setterArgumentType: s, setter: O(o, f), setterContext: c });
    }
    function on(e, r) {
      r = R(r), j(e, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(t, n) {
      } });
    }
    function sn(e) {
      e > 4 && (D.get(e).refcount += 1);
    }
    function un(e, r) {
      var t = z[e];
      return t === void 0 && w(r + " has unknown type " + _r(e)), t;
    }
    function fn(e, r) {
      e = un(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return Ce.toHandle(t);
    }
    function cn() {
      J("");
    }
    function ln(e, r, t) {
      k.copyWithin(e, r, r + t);
    }
    function dn() {
      return 2147483648;
    }
    function vn(e) {
      var r = de.buffer, t = e - r.byteLength + 65535 >>> 16;
      try {
        return de.grow(t), Ge(), 1;
      } catch {
      }
    }
    function pn(e) {
      var r = k.length;
      e = e >>> 0;
      var t = dn();
      if (e > t)
        return !1;
      for (var n = (f, c) => f + (c - f % c) % c, i = 1; i <= 4; i *= 2) {
        var a = r * (1 + 0.2 / i);
        a = Math.min(a, e + 100663296);
        var s = Math.min(t, n(Math.max(e, a), 65536)), o = vn(s);
        if (o)
          return !0;
      }
      return !1;
    }
    var xe = {};
    function hn() {
      return Be || "./this.program";
    }
    function ue() {
      if (!ue.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: hn() };
        for (var t in xe)
          xe[t] === void 0 ? delete r[t] : r[t] = xe[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        ue.strings = n;
      }
      return ue.strings;
    }
    function _n(e, r) {
      for (var t = 0; t < e.length; ++t)
        U[r++ >> 0] = e.charCodeAt(t);
      U[r >> 0] = 0;
    }
    function gn(e, r) {
      var t = 0;
      return ue().forEach(function(n, i) {
        var a = r + t;
        T[e + i * 4 >> 2] = a, _n(n, a), t += n.length + 1;
      }), 0;
    }
    function yn(e, r) {
      var t = ue();
      T[e >> 2] = t.length;
      var n = 0;
      return t.forEach(function(i) {
        n += i.length + 1;
      }), T[r >> 2] = n, 0;
    }
    function mn(e) {
      return e;
    }
    function Pe(e) {
      return e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0);
    }
    function wn(e, r) {
      for (var t = 0, n = 0; n <= r; t += e[n++])
        ;
      return t;
    }
    var $r = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Cr = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    function bn(e, r) {
      for (var t = new Date(e.getTime()); r > 0; ) {
        var n = Pe(t.getFullYear()), i = t.getMonth(), a = (n ? $r : Cr)[i];
        if (r > a - t.getDate())
          r -= a - t.getDate() + 1, t.setDate(1), i < 11 ? t.setMonth(i + 1) : (t.setMonth(0), t.setFullYear(t.getFullYear() + 1));
        else
          return t.setDate(t.getDate() + r), t;
      }
      return t;
    }
    function $n(e, r, t) {
      var n = t > 0 ? t : mr(e) + 1, i = new Array(n), a = yr(e, i, 0, i.length);
      return r && (i.length = a), i;
    }
    function Cn(e, r) {
      U.set(e, r);
    }
    function Pn(e, r, t, n) {
      var i = F[n + 40 >> 2], a = { tm_sec: F[n >> 2], tm_min: F[n + 4 >> 2], tm_hour: F[n + 8 >> 2], tm_mday: F[n + 12 >> 2], tm_mon: F[n + 16 >> 2], tm_year: F[n + 20 >> 2], tm_wday: F[n + 24 >> 2], tm_yday: F[n + 28 >> 2], tm_isdst: F[n + 32 >> 2], tm_gmtoff: F[n + 36 >> 2], tm_zone: i ? Ie(i) : "" }, s = Ie(t), o = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var f in o)
        s = s.replace(new RegExp(f, "g"), o[f]);
      var c = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], d = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function p(l, v, P) {
        for (var A = typeof l == "number" ? l.toString() : l || ""; A.length < v; )
          A = P[0] + A;
        return A;
      }
      function h(l, v) {
        return p(l, v, "0");
      }
      function m(l, v) {
        function P(Z) {
          return Z < 0 ? -1 : Z > 0 ? 1 : 0;
        }
        var A;
        return (A = P(l.getFullYear() - v.getFullYear())) === 0 && (A = P(l.getMonth() - v.getMonth())) === 0 && (A = P(l.getDate() - v.getDate())), A;
      }
      function $(l) {
        switch (l.getDay()) {
          case 0:
            return new Date(l.getFullYear() - 1, 11, 29);
          case 1:
            return l;
          case 2:
            return new Date(l.getFullYear(), 0, 3);
          case 3:
            return new Date(l.getFullYear(), 0, 2);
          case 4:
            return new Date(l.getFullYear(), 0, 1);
          case 5:
            return new Date(l.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(l.getFullYear() - 1, 11, 30);
        }
      }
      function E(l) {
        var v = bn(new Date(l.tm_year + 1900, 0, 1), l.tm_yday), P = new Date(v.getFullYear(), 0, 4), A = new Date(v.getFullYear() + 1, 0, 4), Z = $(P), Ae = $(A);
        return m(Z, v) <= 0 ? m(Ae, v) <= 0 ? v.getFullYear() + 1 : v.getFullYear() : v.getFullYear() - 1;
      }
      var C = { "%a": function(l) {
        return c[l.tm_wday].substring(0, 3);
      }, "%A": function(l) {
        return c[l.tm_wday];
      }, "%b": function(l) {
        return d[l.tm_mon].substring(0, 3);
      }, "%B": function(l) {
        return d[l.tm_mon];
      }, "%C": function(l) {
        var v = l.tm_year + 1900;
        return h(v / 100 | 0, 2);
      }, "%d": function(l) {
        return h(l.tm_mday, 2);
      }, "%e": function(l) {
        return p(l.tm_mday, 2, " ");
      }, "%g": function(l) {
        return E(l).toString().substring(2);
      }, "%G": function(l) {
        return E(l);
      }, "%H": function(l) {
        return h(l.tm_hour, 2);
      }, "%I": function(l) {
        var v = l.tm_hour;
        return v == 0 ? v = 12 : v > 12 && (v -= 12), h(v, 2);
      }, "%j": function(l) {
        return h(l.tm_mday + wn(Pe(l.tm_year + 1900) ? $r : Cr, l.tm_mon - 1), 3);
      }, "%m": function(l) {
        return h(l.tm_mon + 1, 2);
      }, "%M": function(l) {
        return h(l.tm_min, 2);
      }, "%n": function() {
        return `
`;
      }, "%p": function(l) {
        return l.tm_hour >= 0 && l.tm_hour < 12 ? "AM" : "PM";
      }, "%S": function(l) {
        return h(l.tm_sec, 2);
      }, "%t": function() {
        return "	";
      }, "%u": function(l) {
        return l.tm_wday || 7;
      }, "%U": function(l) {
        var v = l.tm_yday + 7 - l.tm_wday;
        return h(Math.floor(v / 7), 2);
      }, "%V": function(l) {
        var v = Math.floor((l.tm_yday + 7 - (l.tm_wday + 6) % 7) / 7);
        if ((l.tm_wday + 371 - l.tm_yday - 2) % 7 <= 2 && v++, v) {
          if (v == 53) {
            var A = (l.tm_wday + 371 - l.tm_yday) % 7;
            A != 4 && (A != 3 || !Pe(l.tm_year)) && (v = 1);
          }
        } else {
          v = 52;
          var P = (l.tm_wday + 7 - l.tm_yday - 1) % 7;
          (P == 4 || P == 5 && Pe(l.tm_year % 400 - 1)) && v++;
        }
        return h(v, 2);
      }, "%w": function(l) {
        return l.tm_wday;
      }, "%W": function(l) {
        var v = l.tm_yday + 7 - (l.tm_wday + 6) % 7;
        return h(Math.floor(v / 7), 2);
      }, "%y": function(l) {
        return (l.tm_year + 1900).toString().substring(2);
      }, "%Y": function(l) {
        return l.tm_year + 1900;
      }, "%z": function(l) {
        var v = l.tm_gmtoff, P = v >= 0;
        return v = Math.abs(v) / 60, v = v / 60 * 100 + v % 60, (P ? "+" : "-") + ("0000" + v).slice(-4);
      }, "%Z": function(l) {
        return l.tm_zone;
      }, "%%": function() {
        return "%";
      } };
      s = s.replace(/%%/g, "\0\0");
      for (var f in C)
        s.includes(f) && (s = s.replace(new RegExp(f, "g"), C[f](a)));
      s = s.replace(/\0\0/g, "%");
      var S = $n(s, !1);
      return S.length > r ? 0 : (Cn(S, e), S.length - 1);
    }
    function Tn(e, r, t, n, i) {
      return Pn(e, r, t, n);
    }
    or = u.InternalError = Re(Error, "InternalError"), ot(), Q = u.BindingError = Re(Error, "BindingError"), $t(), pt(), Rt(), hr = u.UnboundTypeError = Re(Error, "UnboundTypeError"), It();
    var An = { q: zr, v: Xr, a: Gr, h: qr, l: Jr, J: Kr, S: Qr, n: et, _: rt, d: Zr, ma: it, V: at, ca: st, la: Wt, ka: Ut, F: Ht, ba: xt, R: Vt, T: Nt, y: Lt, t: zt, Q: Gt, L: tn, O: nn, na: an, da: on, ha: gr, ia: sn, ja: fn, K: cn, aa: ln, $: pn, Y: gn, Z: yn, B: Ln, P: ei, E: zn, p: Vn, b: Fn, w: Bn, fa: Zn, c: kn, i: jn, j: xn, r: Mn, z: Nn, x: Yn, I: qn, N: Jn, D: Xn, H: ri, W: ni, U: ii, k: Dn, g: Rn, e: Sn, f: En, M: Qn, m: In, o: On, ga: Un, u: Hn, ea: Kn, A: Gn, s: Wn, G: ti, C: mn, X: Tn };
    Lr();
    var I = u._free = function() {
      return (I = u._free = u.asm.qa).apply(null, arguments);
    }, Ye = u._malloc = function() {
      return (Ye = u._malloc = u.asm.ra).apply(null, arguments);
    }, Pr = function() {
      return (Pr = u.asm.ta).apply(null, arguments);
    };
    u.__embind_initialize_bindings = function() {
      return (u.__embind_initialize_bindings = u.asm.ua).apply(null, arguments);
    };
    var _ = function() {
      return (_ = u.asm.va).apply(null, arguments);
    }, fe = function() {
      return (fe = u.asm.wa).apply(null, arguments);
    }, g = function() {
      return (g = u.asm.xa).apply(null, arguments);
    }, y = function() {
      return (y = u.asm.ya).apply(null, arguments);
    }, Tr = function() {
      return (Tr = u.asm.za).apply(null, arguments);
    }, Ar = function() {
      return (Ar = u.asm.Aa).apply(null, arguments);
    }, Fr = function() {
      return (Fr = u.asm.Ba).apply(null, arguments);
    }, Er = function() {
      return (Er = u.asm.Ca).apply(null, arguments);
    };
    u.dynCall_viijii = function() {
      return (u.dynCall_viijii = u.asm.Da).apply(null, arguments);
    };
    var Sr = u.dynCall_jiii = function() {
      return (Sr = u.dynCall_jiii = u.asm.Ea).apply(null, arguments);
    }, Rr = u.dynCall_jiiii = function() {
      return (Rr = u.dynCall_jiiii = u.asm.Fa).apply(null, arguments);
    };
    u.dynCall_iiiiij = function() {
      return (u.dynCall_iiiiij = u.asm.Ga).apply(null, arguments);
    }, u.dynCall_iiiiijj = function() {
      return (u.dynCall_iiiiijj = u.asm.Ha).apply(null, arguments);
    }, u.dynCall_iiiiiijj = function() {
      return (u.dynCall_iiiiiijj = u.asm.Ia).apply(null, arguments);
    };
    function Fn(e, r) {
      var t = g();
      try {
        return b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function En(e, r, t, n) {
      var i = g();
      try {
        b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Sn(e, r, t) {
      var n = g();
      try {
        b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Rn(e, r) {
      var t = g();
      try {
        b(e)(r);
      } catch (n) {
        if (y(t), n !== n + 0)
          throw n;
        _(1, 0);
      }
    }
    function kn(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Dn(e) {
      var r = g();
      try {
        b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function On(e, r, t, n, i, a) {
      var s = g();
      try {
        b(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Wn(e, r, t, n, i, a, s, o, f, c, d) {
      var p = g();
      try {
        b(e)(r, t, n, i, a, s, o, f, c, d);
      } catch (h) {
        if (y(p), h !== h + 0)
          throw h;
        _(1, 0);
      }
    }
    function Un(e, r, t, n, i, a, s) {
      var o = g();
      try {
        b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (y(o), f !== f + 0)
          throw f;
        _(1, 0);
      }
    }
    function Hn(e, r, t, n, i, a, s, o) {
      var f = g();
      try {
        b(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (y(f), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function jn(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Mn(e, r, t, n, i, a) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function In(e, r, t, n, i) {
      var a = g();
      try {
        b(e)(r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function xn(e, r, t, n, i) {
      var a = g();
      try {
        return b(e)(r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    function Yn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (y(o), f !== f + 0)
          throw f;
        _(1, 0);
      }
    }
    function Vn(e) {
      var r = g();
      try {
        return b(e)();
      } catch (t) {
        if (y(r), t !== t + 0)
          throw t;
        _(1, 0);
      }
    }
    function Nn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        return b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (y(o), f !== f + 0)
          throw f;
        _(1, 0);
      }
    }
    function Bn(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function Ln(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function zn(e, r, t, n, i, a) {
      var s = g();
      try {
        return b(e)(r, t, n, i, a);
      } catch (o) {
        if (y(s), o !== o + 0)
          throw o;
        _(1, 0);
      }
    }
    function Xn(e, r, t, n, i, a, s, o, f, c) {
      var d = g();
      try {
        return b(e)(r, t, n, i, a, s, o, f, c);
      } catch (p) {
        if (y(d), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function Zn(e, r, t) {
      var n = g();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (y(n), i !== i + 0)
          throw i;
        _(1, 0);
      }
    }
    function Gn(e, r, t, n, i, a, s, o, f, c) {
      var d = g();
      try {
        b(e)(r, t, n, i, a, s, o, f, c);
      } catch (p) {
        if (y(d), p !== p + 0)
          throw p;
        _(1, 0);
      }
    }
    function qn(e, r, t, n, i, a, s, o) {
      var f = g();
      try {
        return b(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (y(f), c !== c + 0)
          throw c;
        _(1, 0);
      }
    }
    function Jn(e, r, t, n, i, a, s, o, f) {
      var c = g();
      try {
        return b(e)(r, t, n, i, a, s, o, f);
      } catch (d) {
        if (y(c), d !== d + 0)
          throw d;
        _(1, 0);
      }
    }
    function Kn(e, r, t, n, i, a, s, o, f) {
      var c = g();
      try {
        b(e)(r, t, n, i, a, s, o, f);
      } catch (d) {
        if (y(c), d !== d + 0)
          throw d;
        _(1, 0);
      }
    }
    function Qn(e, r, t, n, i, a, s) {
      var o = g();
      try {
        b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (y(o), f !== f + 0)
          throw f;
        _(1, 0);
      }
    }
    function ei(e, r, t, n) {
      var i = g();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ri(e, r, t, n, i, a, s, o, f, c, d, p) {
      var h = g();
      try {
        return b(e)(r, t, n, i, a, s, o, f, c, d, p);
      } catch (m) {
        if (y(h), m !== m + 0)
          throw m;
        _(1, 0);
      }
    }
    function ti(e, r, t, n, i, a, s, o, f, c, d, p, h, m, $, E) {
      var C = g();
      try {
        b(e)(r, t, n, i, a, s, o, f, c, d, p, h, m, $, E);
      } catch (S) {
        if (y(C), S !== S + 0)
          throw S;
        _(1, 0);
      }
    }
    function ni(e, r, t, n) {
      var i = g();
      try {
        return Sr(e, r, t, n);
      } catch (a) {
        if (y(i), a !== a + 0)
          throw a;
        _(1, 0);
      }
    }
    function ii(e, r, t, n, i) {
      var a = g();
      try {
        return Rr(e, r, t, n, i);
      } catch (s) {
        if (y(a), s !== s + 0)
          throw s;
        _(1, 0);
      }
    }
    var Te;
    te = function e() {
      Te || kr(), Te || (te = e);
    };
    function kr() {
      if (B > 0 || (Wr(), B > 0))
        return;
      function e() {
        Te || (Te = !0, u.calledRun = !0, !ze && (Ur(), G(u), u.onRuntimeInitialized && u.onRuntimeInitialized(), Hr()));
      }
      u.setStatus ? (u.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          u.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (u.preInit)
      for (typeof u.preInit == "function" && (u.preInit = [u.preInit]); u.preInit.length > 0; )
        u.preInit.pop()();
    return kr(), N.ready;
  };
})();
function fi(x) {
  return ai(Ve, x);
}
async function ci(x, {
  tryHarder: N = V.tryHarder,
  formats: u = V.formats,
  maxNumberOfSymbols: G = V.maxNumberOfSymbols
} = V) {
  return oi(
    x,
    {
      tryHarder: N,
      formats: u,
      maxNumberOfSymbols: G
    },
    Ve
  );
}
async function li(x, {
  tryHarder: N = V.tryHarder,
  formats: u = V.formats,
  maxNumberOfSymbols: G = V.maxNumberOfSymbols
} = V) {
  return si(
    x,
    {
      tryHarder: N,
      formats: u,
      maxNumberOfSymbols: G
    },
    Ve
  );
}
export {
  pi as ZXING_BARCODE_FORMAT_NAMES,
  hi as ZXING_CHARACTOR_SET_NAMES,
  V as defaultZXingReadOptions,
  _i as defaultZXingWriteOptions,
  fi as getZXingModule,
  gi as purgeZXingModule,
  li as readBarcodeFromImageData,
  ci as readBarcodeFromImageFile,
  yi as setZXingModuleOverrides
};
