// @sec-ant/zxing-wasm/dist/full/index.js
import { g as ui, r as fi, d as N, a as ci, w as li, c as q } from "../core-469ea631.js";
import { Z as mi, b as wi, p as bi, s as $i } from "../core-469ea631.js";
var ke = (() => {
  var W = import.meta.url;
  return function(H = {}) {
    var u = typeof H < "u" ? H : {}, B, J;
    u.ready = new Promise((e, r) => {
      B = e, J = r;
    });
    var ve = Object.assign({}, u), pe = "./this.program", Xe = typeof window == "object", he = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var j = "";
    function Wr(e) {
      return u.locateFile ? u.locateFile(e, j) : j + e;
    }
    var De;
    (Xe || he) && (he ? j = self.location.href : typeof document < "u" && document.currentScript && (j = document.currentScript.src), W && (j = W), j.indexOf("blob:") !== 0 ? j = j.substr(0, j.replace(/[?#].*/, "").lastIndexOf("/") + 1) : j = "", he && (De = (e) => {
      var r = new XMLHttpRequest();
      return r.open("GET", e, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), u.print || console.log.bind(console);
    var ne = u.printErr || console.error.bind(console);
    Object.assign(u, ve), ve = null, u.arguments && u.arguments, u.thisProgram && (pe = u.thisProgram), u.quit && u.quit;
    var ie;
    u.wasmBinary && (ie = u.wasmBinary), u.noExitRuntime, typeof WebAssembly != "object" && Q("no native wasm support detected");
    var _e, Ze = !1;
    function jr(e, r) {
      e || Q(r);
    }
    var U, k, K, ge, S, F, Ge, qe;
    function Je() {
      var e = _e.buffer;
      u.HEAP8 = U = new Int8Array(e), u.HEAP16 = K = new Int16Array(e), u.HEAP32 = S = new Int32Array(e), u.HEAPU8 = k = new Uint8Array(e), u.HEAPU16 = ge = new Uint16Array(e), u.HEAPU32 = F = new Uint32Array(e), u.HEAPF32 = Ge = new Float32Array(e), u.HEAPF64 = qe = new Float64Array(e);
    }
    var Ke, Qe = [], er = [], rr = [];
    function Ur() {
      if (u.preRun)
        for (typeof u.preRun == "function" && (u.preRun = [u.preRun]); u.preRun.length; )
          Ir(u.preRun.shift());
      Oe(Qe);
    }
    function Hr() {
      Oe(er);
    }
    function Mr() {
      if (u.postRun)
        for (typeof u.postRun == "function" && (u.postRun = [u.postRun]); u.postRun.length; )
          Yr(u.postRun.shift());
      Oe(rr);
    }
    function Ir(e) {
      Qe.unshift(e);
    }
    function xr(e) {
      er.unshift(e);
    }
    function Yr(e) {
      rr.unshift(e);
    }
    var z = 0, ae = null;
    function Vr(e) {
      z++, u.monitorRunDependencies && u.monitorRunDependencies(z);
    }
    function Br(e) {
      if (z--, u.monitorRunDependencies && u.monitorRunDependencies(z), z == 0 && ae) {
        var r = ae;
        ae = null, r();
      }
    }
    function Q(e) {
      u.onAbort && u.onAbort(e), e = "Aborted(" + e + ")", ne(e), Ze = !0, e += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(e);
      throw J(r), r;
    }
    var Lr = "data:application/octet-stream;base64,";
    function tr(e) {
      return e.startsWith(Lr);
    }
    var X;
    u.locateFile ? (X = "zxing_full.wasm", tr(X) || (X = Wr(X))) : X = new URL("/full/zxing_full.wasm", self.location).href;
    function nr(e) {
      try {
        if (e == X && ie)
          return new Uint8Array(ie);
        if (De)
          return De(e);
        throw "both async and sync fetching of the wasm failed";
      } catch (r) {
        Q(r);
      }
    }
    function Nr(e) {
      return !ie && (Xe || he) && typeof fetch == "function" ? fetch(e, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + e + "'";
        return r.arrayBuffer();
      }).catch(() => nr(e)) : Promise.resolve().then(() => nr(e));
    }
    function ir(e, r, t) {
      return Nr(e).then((n) => WebAssembly.instantiate(n, r)).then((n) => n).then(t, (n) => {
        ne("failed to asynchronously prepare wasm: " + n), Q(n);
      });
    }
    function zr(e, r, t, n) {
      return !e && typeof WebAssembly.instantiateStreaming == "function" && !tr(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((i) => {
        var a = WebAssembly.instantiateStreaming(i, t);
        return a.then(n, function(s) {
          return ne("wasm streaming compile failed: " + s), ne("falling back to ArrayBuffer instantiation"), ir(r, t, n);
        });
      }) : ir(r, t, n);
    }
    function Xr() {
      var e = { a: Sn };
      function r(n, i) {
        var a = n.exports;
        return u.asm = a, _e = u.asm.pa, Je(), Ke = u.asm.ta, xr(u.asm.qa), Br(), a;
      }
      Vr();
      function t(n) {
        r(n.instance);
      }
      if (u.instantiateWasm)
        try {
          return u.instantiateWasm(e, r);
        } catch (n) {
          ne("Module.instantiateWasm callback failed with error: " + n), J(n);
        }
      return zr(ie, X, e, t).catch(J), {};
    }
    function Oe(e) {
      for (; e.length > 0; )
        e.shift()(u);
    }
    var ye = [], me = 0;
    function Zr(e) {
      var r = new we(e);
      return r.get_caught() || (r.set_caught(!0), me--), r.set_rethrown(!1), ye.push(r), Er(r.excPtr), r.get_exception_ptr();
    }
    var M = 0;
    function Gr() {
      y(0);
      var e = ye.pop();
      Fr(e.excPtr), M = 0;
    }
    function we(e) {
      this.excPtr = e, this.ptr = e - 24, this.set_type = function(r) {
        F[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return F[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        F[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return F[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, U[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return U[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, U[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return U[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, t) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(t);
      }, this.set_adjusted_ptr = function(r) {
        F[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return F[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Rr(this.get_type());
        if (r)
          return F[this.excPtr >> 2];
        var t = this.get_adjusted_ptr();
        return t !== 0 ? t : this.excPtr;
      };
    }
    function qr(e) {
      throw M || (M = e), M;
    }
    function We() {
      var e = M;
      if (!e)
        return de(0), 0;
      var r = new we(e);
      r.set_adjusted_ptr(e);
      var t = r.get_type();
      if (!t)
        return de(0), e;
      for (var n = 0; n < arguments.length; n++) {
        var i = arguments[n];
        if (i === 0 || i === t)
          break;
        var a = r.ptr + 16;
        if (Sr(i, t, a))
          return de(i), e;
      }
      return de(t), e;
    }
    var Jr = We, Kr = We, Qr = We;
    function et(e) {
      var r = new we(e).get_exception_ptr();
      return r;
    }
    function rt() {
      var e = ye.pop();
      e || Q("no exception to throw");
      var r = e.excPtr;
      throw e.get_rethrown() || (ye.push(e), e.set_rethrown(!0), e.set_caught(!1), me++), M = r, M;
    }
    function tt(e, r, t) {
      var n = new we(e);
      throw n.init(r, t), M = e, me++, M;
    }
    function nt() {
      return me;
    }
    var be = {};
    function je(e) {
      for (; e.length; ) {
        var r = e.pop(), t = e.pop();
        t(r);
      }
    }
    function oe(e) {
      return this.fromWireType(S[e >> 2]);
    }
    var ee = {}, Z = {}, $e = {}, it = 48, at = 57;
    function ar(e) {
      if (e === void 0)
        return "_unknown";
      e = e.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = e.charCodeAt(0);
      return r >= it && r <= at ? `_${e}` : e;
    }
    function or(e, r) {
      return e = ar(e), { [e]: function() {
        return r.apply(this, arguments);
      } }[e];
    }
    function Ue(e, r) {
      var t = or(r, function(n) {
        this.name = r, this.message = n;
        var i = new Error(n).stack;
        i !== void 0 && (this.stack = this.toString() + `
` + i.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return t.prototype = Object.create(e.prototype), t.prototype.constructor = t, t.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, t;
    }
    var sr = void 0;
    function Ce(e) {
      throw new sr(e);
    }
    function I(e, r, t) {
      e.forEach(function(o) {
        $e[o] = r;
      });
      function n(o) {
        var f = t(o);
        f.length !== e.length && Ce("Mismatched type converter count");
        for (var c = 0; c < e.length; ++c)
          x(e[c], f[c]);
      }
      var i = new Array(r.length), a = [], s = 0;
      r.forEach((o, f) => {
        Z.hasOwnProperty(o) ? i[f] = Z[o] : (a.push(o), ee.hasOwnProperty(o) || (ee[o] = []), ee[o].push(() => {
          i[f] = Z[o], ++s, s === a.length && n(i);
        }));
      }), a.length === 0 && n(i);
    }
    function ot(e) {
      var r = be[e];
      delete be[e];
      var t = r.rawConstructor, n = r.rawDestructor, i = r.fields, a = i.map((s) => s.getterReturnType).concat(i.map((s) => s.setterArgumentType));
      I([e], a, (s) => {
        var o = {};
        return i.forEach((f, c) => {
          var d = f.fieldName, p = s[c], h = f.getter, _ = f.getterContext, $ = s[c + i.length], T = f.setter, C = f.setterContext;
          o[d] = { read: (P) => p.fromWireType(h(_, P)), write: (P, l) => {
            var v = [];
            T(C, P, $.toWireType(v, l)), je(v);
          } };
        }), [{ name: r.name, fromWireType: function(f) {
          var c = {};
          for (var d in o)
            c[d] = o[d].read(f);
          return n(f), c;
        }, toWireType: function(f, c) {
          for (var d in o)
            if (!(d in c))
              throw new TypeError(`Missing field: "${d}"`);
          var p = t();
          for (d in o)
            o[d].write(p, c[d]);
          return f !== null && f.push(n, p), p;
        }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: n }];
      });
    }
    function st(e, r, t, n, i) {
    }
    function He(e) {
      switch (e) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${e}`);
      }
    }
    function ut() {
      for (var e = new Array(256), r = 0; r < 256; ++r)
        e[r] = String.fromCharCode(r);
      ur = e;
    }
    var ur = void 0;
    function R(e) {
      for (var r = "", t = e; k[t]; )
        r += ur[k[t++]];
      return r;
    }
    var re = void 0;
    function g(e) {
      throw new re(e);
    }
    function x(e, r, t = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      var n = r.name;
      if (e || g(`type "${n}" must have a positive integer typeid pointer`), Z.hasOwnProperty(e)) {
        if (t.ignoreDuplicateRegistrations)
          return;
        g(`Cannot register type '${n}' twice`);
      }
      if (Z[e] = r, delete $e[e], ee.hasOwnProperty(e)) {
        var i = ee[e];
        delete ee[e], i.forEach((a) => a());
      }
    }
    function ft(e, r, t, n, i) {
      var a = He(t);
      r = R(r), x(e, { name: r, fromWireType: function(s) {
        return !!s;
      }, toWireType: function(s, o) {
        return o ? n : i;
      }, argPackAdvance: 8, readValueFromPointer: function(s) {
        var o;
        if (t === 1)
          o = U;
        else if (t === 2)
          o = K;
        else if (t === 4)
          o = S;
        else
          throw new TypeError("Unknown boolean type size: " + r);
        return this.fromWireType(o[s >> a]);
      }, destructorFunction: null });
    }
    function ct(e) {
      if (!(this instanceof L) || !(e instanceof L))
        return !1;
      for (var r = this.$$.ptrType.registeredClass, t = this.$$.ptr, n = e.$$.ptrType.registeredClass, i = e.$$.ptr; r.baseClass; )
        t = r.upcast(t), r = r.baseClass;
      for (; n.baseClass; )
        i = n.upcast(i), n = n.baseClass;
      return r === n && t === i;
    }
    function lt(e) {
      return { count: e.count, deleteScheduled: e.deleteScheduled, preservePointerOnDelete: e.preservePointerOnDelete, ptr: e.ptr, ptrType: e.ptrType, smartPtr: e.smartPtr, smartPtrType: e.smartPtrType };
    }
    function Me(e) {
      function r(t) {
        return t.$$.ptrType.registeredClass.name;
      }
      g(r(e) + " instance already deleted");
    }
    var Ie = !1;
    function fr(e) {
    }
    function dt(e) {
      e.smartPtr ? e.smartPtrType.rawDestructor(e.smartPtr) : e.ptrType.registeredClass.rawDestructor(e.ptr);
    }
    function cr(e) {
      e.count.value -= 1;
      var r = e.count.value === 0;
      r && dt(e);
    }
    function lr(e, r, t) {
      if (r === t)
        return e;
      if (t.baseClass === void 0)
        return null;
      var n = lr(e, r, t.baseClass);
      return n === null ? null : t.downcast(n);
    }
    var dr = {};
    function vt() {
      return Object.keys(fe).length;
    }
    function pt() {
      var e = [];
      for (var r in fe)
        fe.hasOwnProperty(r) && e.push(fe[r]);
      return e;
    }
    var se = [];
    function xe() {
      for (; se.length; ) {
        var e = se.pop();
        e.$$.deleteScheduled = !1, e.delete();
      }
    }
    var ue = void 0;
    function ht(e) {
      ue = e, se.length && ue && ue(xe);
    }
    function _t() {
      u.getInheritedInstanceCount = vt, u.getLiveInheritedInstances = pt, u.flushPendingDeletes = xe, u.setDelayFunction = ht;
    }
    var fe = {};
    function gt(e, r) {
      for (r === void 0 && g("ptr should not be undefined"); e.baseClass; )
        r = e.upcast(r), e = e.baseClass;
      return r;
    }
    function yt(e, r) {
      return r = gt(e, r), fe[r];
    }
    function Pe(e, r) {
      (!r.ptrType || !r.ptr) && Ce("makeClassHandle requires ptr and ptrType");
      var t = !!r.smartPtrType, n = !!r.smartPtr;
      return t !== n && Ce("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, ce(Object.create(e, { $$: { value: r } }));
    }
    function mt(e) {
      var r = this.getPointee(e);
      if (!r)
        return this.destructor(e), null;
      var t = yt(this.registeredClass, r);
      if (t !== void 0) {
        if (t.$$.count.value === 0)
          return t.$$.ptr = r, t.$$.smartPtr = e, t.clone();
        var n = t.clone();
        return this.destructor(e), n;
      }
      function i() {
        return this.isSmartPointer ? Pe(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: e }) : Pe(this.registeredClass.instancePrototype, { ptrType: this, ptr: e });
      }
      var a = this.registeredClass.getActualType(r), s = dr[a];
      if (!s)
        return i.call(this);
      var o;
      this.isConst ? o = s.constPointerType : o = s.pointerType;
      var f = lr(r, this.registeredClass, o.registeredClass);
      return f === null ? i.call(this) : this.isSmartPointer ? Pe(o.registeredClass.instancePrototype, { ptrType: o, ptr: f, smartPtrType: this, smartPtr: e }) : Pe(o.registeredClass.instancePrototype, { ptrType: o, ptr: f });
    }
    function ce(e) {
      return typeof FinalizationRegistry > "u" ? (ce = (r) => r, e) : (Ie = new FinalizationRegistry((r) => {
        cr(r.$$);
      }), ce = (r) => {
        var t = r.$$, n = !!t.smartPtr;
        if (n) {
          var i = { $$: t };
          Ie.register(r, i, r);
        }
        return r;
      }, fr = (r) => Ie.unregister(r), ce(e));
    }
    function wt() {
      if (this.$$.ptr || Me(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var e = ce(Object.create(Object.getPrototypeOf(this), { $$: { value: lt(this.$$) } }));
      return e.$$.count.value += 1, e.$$.deleteScheduled = !1, e;
    }
    function bt() {
      this.$$.ptr || Me(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && g("Object already scheduled for deletion"), fr(this), cr(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function $t() {
      return !this.$$.ptr;
    }
    function Ct() {
      return this.$$.ptr || Me(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && g("Object already scheduled for deletion"), se.push(this), se.length === 1 && ue && ue(xe), this.$$.deleteScheduled = !0, this;
    }
    function Pt() {
      L.prototype.isAliasOf = ct, L.prototype.clone = wt, L.prototype.delete = bt, L.prototype.isDeleted = $t, L.prototype.deleteLater = Ct;
    }
    function L() {
    }
    function vr(e, r, t) {
      if (e[r].overloadTable === void 0) {
        var n = e[r];
        e[r] = function() {
          return e[r].overloadTable.hasOwnProperty(arguments.length) || g(`Function '${t}' called with an invalid number of arguments (${arguments.length}) - expects one of (${e[r].overloadTable})!`), e[r].overloadTable[arguments.length].apply(this, arguments);
        }, e[r].overloadTable = [], e[r].overloadTable[n.argCount] = n;
      }
    }
    function pr(e, r, t) {
      u.hasOwnProperty(e) ? ((t === void 0 || u[e].overloadTable !== void 0 && u[e].overloadTable[t] !== void 0) && g(`Cannot register public name '${e}' twice`), vr(u, e, e), u.hasOwnProperty(t) && g(`Cannot register multiple overloads of a function with the same number of arguments (${t})!`), u[e].overloadTable[t] = r) : (u[e] = r, t !== void 0 && (u[e].numArguments = t));
    }
    function Tt(e, r, t, n, i, a, s, o) {
      this.name = e, this.constructor = r, this.instancePrototype = t, this.rawDestructor = n, this.baseClass = i, this.getActualType = a, this.upcast = s, this.downcast = o, this.pureVirtualFunctions = [];
    }
    function Te(e, r, t) {
      for (; r !== t; )
        r.upcast || g(`Expected null or instance of ${t.name}, got an instance of ${r.name}`), e = r.upcast(e), r = r.baseClass;
      return e;
    }
    function At(e, r) {
      if (r === null)
        return this.isReference && g(`null is not a valid ${this.name}`), 0;
      r.$$ || g(`Cannot pass "${Be(r)}" as a ${this.name}`), r.$$.ptr || g(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Te(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function Ft(e, r) {
      var t;
      if (r === null)
        return this.isReference && g(`null is not a valid ${this.name}`), this.isSmartPointer ? (t = this.rawConstructor(), e !== null && e.push(this.rawDestructor, t), t) : 0;
      r.$$ || g(`Cannot pass "${Be(r)}" as a ${this.name}`), r.$$.ptr || g(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && g(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass;
      if (t = Te(r.$$.ptr, n, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && g("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? t = r.$$.smartPtr : g(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            t = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              t = r.$$.smartPtr;
            else {
              var i = r.clone();
              t = this.rawShare(t, Fe.toHandle(function() {
                i.delete();
              })), e !== null && e.push(this.rawDestructor, t);
            }
            break;
          default:
            g("Unsupporting sharing policy");
        }
      return t;
    }
    function Et(e, r) {
      if (r === null)
        return this.isReference && g(`null is not a valid ${this.name}`), 0;
      r.$$ || g(`Cannot pass "${Be(r)}" as a ${this.name}`), r.$$.ptr || g(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && g(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var t = r.$$.ptrType.registeredClass, n = Te(r.$$.ptr, t, this.registeredClass);
      return n;
    }
    function St(e) {
      return this.rawGetPointee && (e = this.rawGetPointee(e)), e;
    }
    function Rt(e) {
      this.rawDestructor && this.rawDestructor(e);
    }
    function kt(e) {
      e !== null && e.delete();
    }
    function Dt() {
      Y.prototype.getPointee = St, Y.prototype.destructor = Rt, Y.prototype.argPackAdvance = 8, Y.prototype.readValueFromPointer = oe, Y.prototype.deleteObject = kt, Y.prototype.fromWireType = mt;
    }
    function Y(e, r, t, n, i, a, s, o, f, c, d) {
      this.name = e, this.registeredClass = r, this.isReference = t, this.isConst = n, this.isSmartPointer = i, this.pointeeType = a, this.sharingPolicy = s, this.rawGetPointee = o, this.rawConstructor = f, this.rawShare = c, this.rawDestructor = d, !i && r.baseClass === void 0 ? n ? (this.toWireType = At, this.destructorFunction = null) : (this.toWireType = Et, this.destructorFunction = null) : this.toWireType = Ft;
    }
    function hr(e, r, t) {
      u.hasOwnProperty(e) || Ce("Replacing nonexistant public symbol"), u[e].overloadTable !== void 0 && t !== void 0 ? u[e].overloadTable[t] = r : (u[e] = r, u[e].argCount = t);
    }
    function Ot(e, r, t) {
      var n = u["dynCall_" + e];
      return t && t.length ? n.apply(null, [r].concat(t)) : n.call(null, r);
    }
    var Ae = [];
    function b(e) {
      var r = Ae[e];
      return r || (e >= Ae.length && (Ae.length = e + 1), Ae[e] = r = Ke.get(e)), r;
    }
    function Wt(e, r, t) {
      if (e.includes("j"))
        return Ot(e, r, t);
      var n = b(r).apply(null, t);
      return n;
    }
    function jt(e, r) {
      var t = [];
      return function() {
        return t.length = 0, Object.assign(t, arguments), Wt(e, r, t);
      };
    }
    function D(e, r) {
      e = R(e);
      function t() {
        return e.includes("j") ? jt(e, r) : b(r);
      }
      var n = t();
      return typeof n != "function" && g(`unknown function pointer with signature ${e}: ${r}`), n;
    }
    var _r = void 0;
    function gr(e) {
      var r = Ar(e), t = R(r);
      return V(r), t;
    }
    function te(e, r) {
      var t = [], n = {};
      function i(a) {
        if (!n[a] && !Z[a]) {
          if ($e[a]) {
            $e[a].forEach(i);
            return;
          }
          t.push(a), n[a] = !0;
        }
      }
      throw r.forEach(i), new _r(`${e}: ` + t.map(gr).join([", "]));
    }
    function Ut(e, r, t, n, i, a, s, o, f, c, d, p, h) {
      d = R(d), a = D(i, a), o && (o = D(s, o)), c && (c = D(f, c)), h = D(p, h);
      var _ = ar(d);
      pr(_, function() {
        te(`Cannot construct ${d} due to unbound types`, [n]);
      }), I([e, r, t], n ? [n] : [], function($) {
        $ = $[0];
        var T, C;
        n ? (T = $.registeredClass, C = T.instancePrototype) : C = L.prototype;
        var P = or(_, function() {
          if (Object.getPrototypeOf(this) !== l)
            throw new re("Use 'new' to construct " + d);
          if (v.constructor_body === void 0)
            throw new re(d + " has no accessible constructor");
          var Re = v.constructor_body[arguments.length];
          if (Re === void 0)
            throw new re(`Tried to invoke ctor of ${d} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(v.constructor_body).toString()}) parameters instead!`);
          return Re.apply(this, arguments);
        }), l = Object.create(C, { constructor: { value: P } });
        P.prototype = l;
        var v = new Tt(d, P, l, h, T, a, o, c);
        v.baseClass && (v.baseClass.__derivedClasses === void 0 && (v.baseClass.__derivedClasses = []), v.baseClass.__derivedClasses.push(v));
        var A = new Y(d, v, !0, !1, !1), E = new Y(d + "*", v, !1, !1, !1), G = new Y(d + " const*", v, !1, !0, !1);
        return dr[e] = { pointerType: E, constPointerType: G }, hr(_, P), [A, E, G];
      });
    }
    function Ye(e, r) {
      for (var t = [], n = 0; n < e; n++)
        t.push(F[r + n * 4 >> 2]);
      return t;
    }
    function Ve(e, r, t, n, i, a) {
      var s = r.length;
      s < 2 && g("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var o = r[1] !== null && t !== null, f = !1, c = 1; c < r.length; ++c)
        if (r[c] !== null && r[c].destructorFunction === void 0) {
          f = !0;
          break;
        }
      var d = r[0].name !== "void", p = s - 2, h = new Array(p), _ = [], $ = [];
      return function() {
        arguments.length !== p && g(`function ${e} called with ${arguments.length} arguments, expected ${p} args!`), $.length = 0;
        var T;
        _.length = o ? 2 : 1, _[0] = i, o && (T = r[1].toWireType($, this), _[1] = T);
        for (var C = 0; C < p; ++C)
          h[C] = r[C + 2].toWireType($, arguments[C]), _.push(h[C]);
        var P = n.apply(null, _);
        function l(v) {
          if (f)
            je($);
          else
            for (var A = o ? 1 : 2; A < r.length; A++) {
              var E = A === 1 ? T : h[A - 2];
              r[A].destructorFunction !== null && r[A].destructorFunction(E);
            }
          if (d)
            return r[0].fromWireType(v);
        }
        return l(P);
      };
    }
    function Ht(e, r, t, n, i, a) {
      jr(r > 0);
      var s = Ye(r, t);
      i = D(n, i), I([], [e], function(o) {
        o = o[0];
        var f = `constructor ${o.name}`;
        if (o.registeredClass.constructor_body === void 0 && (o.registeredClass.constructor_body = []), o.registeredClass.constructor_body[r - 1] !== void 0)
          throw new re(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${o.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return o.registeredClass.constructor_body[r - 1] = () => {
          te(`Cannot construct ${o.name} due to unbound types`, s);
        }, I([], s, function(c) {
          return c.splice(1, 0, null), o.registeredClass.constructor_body[r - 1] = Ve(f, c, null, i, a), [];
        }), [];
      });
    }
    function Mt(e, r, t, n, i, a, s, o, f) {
      var c = Ye(t, n);
      r = R(r), a = D(i, a), I([], [e], function(d) {
        d = d[0];
        var p = `${d.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), o && d.registeredClass.pureVirtualFunctions.push(r);
        function h() {
          te(`Cannot call ${p} due to unbound types`, c);
        }
        var _ = d.registeredClass.instancePrototype, $ = _[r];
        return $ === void 0 || $.overloadTable === void 0 && $.className !== d.name && $.argCount === t - 2 ? (h.argCount = t - 2, h.className = d.name, _[r] = h) : (vr(_, r, p), _[r].overloadTable[t - 2] = h), I([], c, function(T) {
          var C = Ve(p, T, d, a, s);
          return _[r].overloadTable === void 0 ? (C.argCount = t - 2, _[r] = C) : _[r].overloadTable[t - 2] = C, [];
        }), [];
      });
    }
    function yr(e, r, t) {
      return e instanceof Object || g(`${t} with invalid "this": ${e}`), e instanceof r.registeredClass.constructor || g(`${t} incompatible with "this" of type ${e.constructor.name}`), e.$$.ptr || g(`cannot call emscripten binding method ${t} on deleted object`), Te(e.$$.ptr, e.$$.ptrType.registeredClass, r.registeredClass);
    }
    function It(e, r, t, n, i, a, s, o, f, c) {
      r = R(r), i = D(n, i), I([], [e], function(d) {
        d = d[0];
        var p = `${d.name}.${r}`, h = { get: function() {
          te(`Cannot access ${p} due to unbound types`, [t, s]);
        }, enumerable: !0, configurable: !0 };
        return f ? h.set = () => {
          te(`Cannot access ${p} due to unbound types`, [t, s]);
        } : h.set = (_) => {
          g(p + " is a read-only property");
        }, Object.defineProperty(d.registeredClass.instancePrototype, r, h), I([], f ? [t, s] : [t], function(_) {
          var $ = _[0], T = { get: function() {
            var P = yr(this, d, p + " getter");
            return $.fromWireType(i(a, P));
          }, enumerable: !0 };
          if (f) {
            f = D(o, f);
            var C = _[1];
            T.set = function(P) {
              var l = yr(this, d, p + " setter"), v = [];
              f(c, l, C.toWireType(v, P)), je(v);
            };
          }
          return Object.defineProperty(d.registeredClass.instancePrototype, r, T), [];
        }), [];
      });
    }
    function xt() {
      this.allocated = [void 0], this.freelist = [], this.get = function(e) {
        return this.allocated[e];
      }, this.has = function(e) {
        return this.allocated[e] !== void 0;
      }, this.allocate = function(e) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = e, r;
      }, this.free = function(e) {
        this.allocated[e] = void 0, this.freelist.push(e);
      };
    }
    var O = new xt();
    function mr(e) {
      e >= O.reserved && --O.get(e).refcount === 0 && O.free(e);
    }
    function Yt() {
      for (var e = 0, r = O.reserved; r < O.allocated.length; ++r)
        O.allocated[r] !== void 0 && ++e;
      return e;
    }
    function Vt() {
      O.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), O.reserved = O.allocated.length, u.count_emval_handles = Yt;
    }
    var Fe = { toValue: (e) => (e || g("Cannot use deleted val. handle = " + e), O.get(e).value), toHandle: (e) => {
      switch (e) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return O.allocate({ refcount: 1, value: e });
      }
    } };
    function Bt(e, r) {
      r = R(r), x(e, { name: r, fromWireType: function(t) {
        var n = Fe.toValue(t);
        return mr(t), n;
      }, toWireType: function(t, n) {
        return Fe.toHandle(n);
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: null });
    }
    function Be(e) {
      if (e === null)
        return "null";
      var r = typeof e;
      return r === "object" || r === "array" || r === "function" ? e.toString() : "" + e;
    }
    function Lt(e, r) {
      switch (r) {
        case 2:
          return function(t) {
            return this.fromWireType(Ge[t >> 2]);
          };
        case 3:
          return function(t) {
            return this.fromWireType(qe[t >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + e);
      }
    }
    function Nt(e, r, t) {
      var n = He(t);
      r = R(r), x(e, { name: r, fromWireType: function(i) {
        return i;
      }, toWireType: function(i, a) {
        return a;
      }, argPackAdvance: 8, readValueFromPointer: Lt(r, n), destructorFunction: null });
    }
    function zt(e, r, t, n, i, a, s) {
      var o = Ye(r, t);
      e = R(e), i = D(n, i), pr(e, function() {
        te(`Cannot call ${e} due to unbound types`, o);
      }, r - 1), I([], o, function(f) {
        var c = [f[0], null].concat(f.slice(1));
        return hr(e, Ve(e, c, null, i, a), r - 1), [];
      });
    }
    function Xt(e, r, t) {
      switch (r) {
        case 0:
          return t ? function(i) {
            return U[i];
          } : function(i) {
            return k[i];
          };
        case 1:
          return t ? function(i) {
            return K[i >> 1];
          } : function(i) {
            return ge[i >> 1];
          };
        case 2:
          return t ? function(i) {
            return S[i >> 2];
          } : function(i) {
            return F[i >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + e);
      }
    }
    function Zt(e, r, t, n, i) {
      r = R(r);
      var a = He(t), s = (p) => p;
      if (n === 0) {
        var o = 32 - 8 * t;
        s = (p) => p << o >>> o;
      }
      var f = r.includes("unsigned"), c = (p, h) => {
      }, d;
      f ? d = function(p, h) {
        return c(h, this.name), h >>> 0;
      } : d = function(p, h) {
        return c(h, this.name), h;
      }, x(e, { name: r, fromWireType: s, toWireType: d, argPackAdvance: 8, readValueFromPointer: Xt(r, a, n !== 0), destructorFunction: null });
    }
    function Gt(e, r, t) {
      var n = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], i = n[r];
      function a(s) {
        s = s >> 2;
        var o = F, f = o[s], c = o[s + 1];
        return new i(o.buffer, c, f);
      }
      t = R(t), x(e, { name: t, fromWireType: a, argPackAdvance: 8, readValueFromPointer: a }, { ignoreDuplicateRegistrations: !0 });
    }
    function wr(e, r, t, n) {
      if (!(n > 0))
        return 0;
      for (var i = t, a = t + n - 1, s = 0; s < e.length; ++s) {
        var o = e.charCodeAt(s);
        if (o >= 55296 && o <= 57343) {
          var f = e.charCodeAt(++s);
          o = 65536 + ((o & 1023) << 10) | f & 1023;
        }
        if (o <= 127) {
          if (t >= a)
            break;
          r[t++] = o;
        } else if (o <= 2047) {
          if (t + 1 >= a)
            break;
          r[t++] = 192 | o >> 6, r[t++] = 128 | o & 63;
        } else if (o <= 65535) {
          if (t + 2 >= a)
            break;
          r[t++] = 224 | o >> 12, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        } else {
          if (t + 3 >= a)
            break;
          r[t++] = 240 | o >> 18, r[t++] = 128 | o >> 12 & 63, r[t++] = 128 | o >> 6 & 63, r[t++] = 128 | o & 63;
        }
      }
      return r[t] = 0, t - i;
    }
    function qt(e, r, t) {
      return wr(e, k, r, t);
    }
    function br(e) {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n <= 127 ? r++ : n <= 2047 ? r += 2 : n >= 55296 && n <= 57343 ? (r += 4, ++t) : r += 3;
      }
      return r;
    }
    var $r = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0;
    function Jt(e, r, t) {
      for (var n = r + t, i = r; e[i] && !(i >= n); )
        ++i;
      if (i - r > 16 && e.buffer && $r)
        return $r.decode(e.subarray(r, i));
      for (var a = ""; r < i; ) {
        var s = e[r++];
        if (!(s & 128)) {
          a += String.fromCharCode(s);
          continue;
        }
        var o = e[r++] & 63;
        if ((s & 224) == 192) {
          a += String.fromCharCode((s & 31) << 6 | o);
          continue;
        }
        var f = e[r++] & 63;
        if ((s & 240) == 224 ? s = (s & 15) << 12 | o << 6 | f : s = (s & 7) << 18 | o << 12 | f << 6 | e[r++] & 63, s < 65536)
          a += String.fromCharCode(s);
        else {
          var c = s - 65536;
          a += String.fromCharCode(55296 | c >> 10, 56320 | c & 1023);
        }
      }
      return a;
    }
    function Le(e, r) {
      return e ? Jt(k, e, r) : "";
    }
    function Kt(e, r) {
      r = R(r);
      var t = r === "std::string";
      x(e, { name: r, fromWireType: function(n) {
        var i = F[n >> 2], a = n + 4, s;
        if (t)
          for (var o = a, f = 0; f <= i; ++f) {
            var c = a + f;
            if (f == i || k[c] == 0) {
              var d = c - o, p = Le(o, d);
              s === void 0 ? s = p : (s += String.fromCharCode(0), s += p), o = c + 1;
            }
          }
        else {
          for (var h = new Array(i), f = 0; f < i; ++f)
            h[f] = String.fromCharCode(k[a + f]);
          s = h.join("");
        }
        return V(n), s;
      }, toWireType: function(n, i) {
        i instanceof ArrayBuffer && (i = new Uint8Array(i));
        var a, s = typeof i == "string";
        s || i instanceof Uint8Array || i instanceof Uint8ClampedArray || i instanceof Int8Array || g("Cannot pass non-string to std::string"), t && s ? a = br(i) : a = i.length;
        var o = ze(4 + a + 1), f = o + 4;
        if (F[o >> 2] = a, t && s)
          qt(i, f, a + 1);
        else if (s)
          for (var c = 0; c < a; ++c) {
            var d = i.charCodeAt(c);
            d > 255 && (V(f), g("String has UTF-16 code units that do not fit in 8 bits")), k[f + c] = d;
          }
        else
          for (var c = 0; c < a; ++c)
            k[f + c] = i[c];
        return n !== null && n.push(V, o), o;
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: function(n) {
        V(n);
      } });
    }
    var Cr = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0;
    function Qt(e, r) {
      for (var t = e, n = t >> 1, i = n + r / 2; !(n >= i) && ge[n]; )
        ++n;
      if (t = n << 1, t - e > 32 && Cr)
        return Cr.decode(k.subarray(e, t));
      for (var a = "", s = 0; !(s >= r / 2); ++s) {
        var o = K[e + s * 2 >> 1];
        if (o == 0)
          break;
        a += String.fromCharCode(o);
      }
      return a;
    }
    function en(e, r, t) {
      if (t === void 0 && (t = 2147483647), t < 2)
        return 0;
      t -= 2;
      for (var n = r, i = t < e.length * 2 ? t / 2 : e.length, a = 0; a < i; ++a) {
        var s = e.charCodeAt(a);
        K[r >> 1] = s, r += 2;
      }
      return K[r >> 1] = 0, r - n;
    }
    function rn(e) {
      return e.length * 2;
    }
    function tn(e, r) {
      for (var t = 0, n = ""; !(t >= r / 4); ) {
        var i = S[e + t * 4 >> 2];
        if (i == 0)
          break;
        if (++t, i >= 65536) {
          var a = i - 65536;
          n += String.fromCharCode(55296 | a >> 10, 56320 | a & 1023);
        } else
          n += String.fromCharCode(i);
      }
      return n;
    }
    function nn(e, r, t) {
      if (t === void 0 && (t = 2147483647), t < 4)
        return 0;
      for (var n = r, i = n + t - 4, a = 0; a < e.length; ++a) {
        var s = e.charCodeAt(a);
        if (s >= 55296 && s <= 57343) {
          var o = e.charCodeAt(++a);
          s = 65536 + ((s & 1023) << 10) | o & 1023;
        }
        if (S[r >> 2] = s, r += 4, r + 4 > i)
          break;
      }
      return S[r >> 2] = 0, r - n;
    }
    function an(e) {
      for (var r = 0, t = 0; t < e.length; ++t) {
        var n = e.charCodeAt(t);
        n >= 55296 && n <= 57343 && ++t, r += 4;
      }
      return r;
    }
    function on(e, r, t) {
      t = R(t);
      var n, i, a, s, o;
      r === 2 ? (n = Qt, i = en, s = rn, a = () => ge, o = 1) : r === 4 && (n = tn, i = nn, s = an, a = () => F, o = 2), x(e, { name: t, fromWireType: function(f) {
        for (var c = F[f >> 2], d = a(), p, h = f + 4, _ = 0; _ <= c; ++_) {
          var $ = f + 4 + _ * r;
          if (_ == c || d[$ >> o] == 0) {
            var T = $ - h, C = n(h, T);
            p === void 0 ? p = C : (p += String.fromCharCode(0), p += C), h = $ + r;
          }
        }
        return V(f), p;
      }, toWireType: function(f, c) {
        typeof c != "string" && g(`Cannot pass non-string to C++ string type ${t}`);
        var d = s(c), p = ze(4 + d + r);
        return F[p >> 2] = d >> o, i(c, p + 4, d + r), f !== null && f.push(V, p), p;
      }, argPackAdvance: 8, readValueFromPointer: oe, destructorFunction: function(f) {
        V(f);
      } });
    }
    function sn(e, r, t, n, i, a) {
      be[e] = { name: R(r), rawConstructor: D(t, n), rawDestructor: D(i, a), fields: [] };
    }
    function un(e, r, t, n, i, a, s, o, f, c) {
      be[e].fields.push({ fieldName: R(r), getterReturnType: t, getter: D(n, i), getterContext: a, setterArgumentType: s, setter: D(o, f), setterContext: c });
    }
    function fn(e, r) {
      r = R(r), x(e, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(t, n) {
      } });
    }
    function cn(e) {
      e > 4 && (O.get(e).refcount += 1);
    }
    function ln(e, r) {
      var t = Z[e];
      return t === void 0 && g(r + " has unknown type " + gr(e)), t;
    }
    function dn(e, r) {
      e = ln(e, "_emval_take_value");
      var t = e.readValueFromPointer(r);
      return Fe.toHandle(t);
    }
    function vn() {
      Q("");
    }
    function pn(e, r, t) {
      k.copyWithin(e, r, r + t);
    }
    function hn() {
      return 2147483648;
    }
    function _n(e) {
      var r = _e.buffer, t = e - r.byteLength + 65535 >>> 16;
      try {
        return _e.grow(t), Je(), 1;
      } catch {
      }
    }
    function gn(e) {
      var r = k.length;
      e = e >>> 0;
      var t = hn();
      if (e > t)
        return !1;
      for (var n = (f, c) => f + (c - f % c) % c, i = 1; i <= 4; i *= 2) {
        var a = r * (1 + 0.2 / i);
        a = Math.min(a, e + 100663296);
        var s = Math.min(t, n(Math.max(e, a), 65536)), o = _n(s);
        if (o)
          return !0;
      }
      return !1;
    }
    var Ne = {};
    function yn() {
      return pe || "./this.program";
    }
    function le() {
      if (!le.strings) {
        var e = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: e, _: yn() };
        for (var t in Ne)
          Ne[t] === void 0 ? delete r[t] : r[t] = Ne[t];
        var n = [];
        for (var t in r)
          n.push(`${t}=${r[t]}`);
        le.strings = n;
      }
      return le.strings;
    }
    function mn(e, r) {
      for (var t = 0; t < e.length; ++t)
        U[r++ >> 0] = e.charCodeAt(t);
      U[r >> 0] = 0;
    }
    function wn(e, r) {
      var t = 0;
      return le().forEach(function(n, i) {
        var a = r + t;
        F[e + i * 4 >> 2] = a, mn(n, a), t += n.length + 1;
      }), 0;
    }
    function bn(e, r) {
      var t = le();
      F[e >> 2] = t.length;
      var n = 0;
      return t.forEach(function(i) {
        n += i.length + 1;
      }), F[r >> 2] = n, 0;
    }
    function $n(e) {
      return e;
    }
    function Ee(e) {
      return e % 4 === 0 && (e % 100 !== 0 || e % 400 === 0);
    }
    function Cn(e, r) {
      for (var t = 0, n = 0; n <= r; t += e[n++])
        ;
      return t;
    }
    var Pr = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Tr = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    function Pn(e, r) {
      for (var t = new Date(e.getTime()); r > 0; ) {
        var n = Ee(t.getFullYear()), i = t.getMonth(), a = (n ? Pr : Tr)[i];
        if (r > a - t.getDate())
          r -= a - t.getDate() + 1, t.setDate(1), i < 11 ? t.setMonth(i + 1) : (t.setMonth(0), t.setFullYear(t.getFullYear() + 1));
        else
          return t.setDate(t.getDate() + r), t;
      }
      return t;
    }
    function Tn(e, r, t) {
      var n = t > 0 ? t : br(e) + 1, i = new Array(n), a = wr(e, i, 0, i.length);
      return r && (i.length = a), i;
    }
    function An(e, r) {
      U.set(e, r);
    }
    function Fn(e, r, t, n) {
      var i = S[n + 40 >> 2], a = { tm_sec: S[n >> 2], tm_min: S[n + 4 >> 2], tm_hour: S[n + 8 >> 2], tm_mday: S[n + 12 >> 2], tm_mon: S[n + 16 >> 2], tm_year: S[n + 20 >> 2], tm_wday: S[n + 24 >> 2], tm_yday: S[n + 28 >> 2], tm_isdst: S[n + 32 >> 2], tm_gmtoff: S[n + 36 >> 2], tm_zone: i ? Le(i) : "" }, s = Le(t), o = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var f in o)
        s = s.replace(new RegExp(f, "g"), o[f]);
      var c = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], d = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function p(l, v, A) {
        for (var E = typeof l == "number" ? l.toString() : l || ""; E.length < v; )
          E = A[0] + E;
        return E;
      }
      function h(l, v) {
        return p(l, v, "0");
      }
      function _(l, v) {
        function A(G) {
          return G < 0 ? -1 : G > 0 ? 1 : 0;
        }
        var E;
        return (E = A(l.getFullYear() - v.getFullYear())) === 0 && (E = A(l.getMonth() - v.getMonth())) === 0 && (E = A(l.getDate() - v.getDate())), E;
      }
      function $(l) {
        switch (l.getDay()) {
          case 0:
            return new Date(l.getFullYear() - 1, 11, 29);
          case 1:
            return l;
          case 2:
            return new Date(l.getFullYear(), 0, 3);
          case 3:
            return new Date(l.getFullYear(), 0, 2);
          case 4:
            return new Date(l.getFullYear(), 0, 1);
          case 5:
            return new Date(l.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(l.getFullYear() - 1, 11, 30);
        }
      }
      function T(l) {
        var v = Pn(new Date(l.tm_year + 1900, 0, 1), l.tm_yday), A = new Date(v.getFullYear(), 0, 4), E = new Date(v.getFullYear() + 1, 0, 4), G = $(A), Re = $(E);
        return _(G, v) <= 0 ? _(Re, v) <= 0 ? v.getFullYear() + 1 : v.getFullYear() : v.getFullYear() - 1;
      }
      var C = { "%a": function(l) {
        return c[l.tm_wday].substring(0, 3);
      }, "%A": function(l) {
        return c[l.tm_wday];
      }, "%b": function(l) {
        return d[l.tm_mon].substring(0, 3);
      }, "%B": function(l) {
        return d[l.tm_mon];
      }, "%C": function(l) {
        var v = l.tm_year + 1900;
        return h(v / 100 | 0, 2);
      }, "%d": function(l) {
        return h(l.tm_mday, 2);
      }, "%e": function(l) {
        return p(l.tm_mday, 2, " ");
      }, "%g": function(l) {
        return T(l).toString().substring(2);
      }, "%G": function(l) {
        return T(l);
      }, "%H": function(l) {
        return h(l.tm_hour, 2);
      }, "%I": function(l) {
        var v = l.tm_hour;
        return v == 0 ? v = 12 : v > 12 && (v -= 12), h(v, 2);
      }, "%j": function(l) {
        return h(l.tm_mday + Cn(Ee(l.tm_year + 1900) ? Pr : Tr, l.tm_mon - 1), 3);
      }, "%m": function(l) {
        return h(l.tm_mon + 1, 2);
      }, "%M": function(l) {
        return h(l.tm_min, 2);
      }, "%n": function() {
        return `
`;
      }, "%p": function(l) {
        return l.tm_hour >= 0 && l.tm_hour < 12 ? "AM" : "PM";
      }, "%S": function(l) {
        return h(l.tm_sec, 2);
      }, "%t": function() {
        return "	";
      }, "%u": function(l) {
        return l.tm_wday || 7;
      }, "%U": function(l) {
        var v = l.tm_yday + 7 - l.tm_wday;
        return h(Math.floor(v / 7), 2);
      }, "%V": function(l) {
        var v = Math.floor((l.tm_yday + 7 - (l.tm_wday + 6) % 7) / 7);
        if ((l.tm_wday + 371 - l.tm_yday - 2) % 7 <= 2 && v++, v) {
          if (v == 53) {
            var E = (l.tm_wday + 371 - l.tm_yday) % 7;
            E != 4 && (E != 3 || !Ee(l.tm_year)) && (v = 1);
          }
        } else {
          v = 52;
          var A = (l.tm_wday + 7 - l.tm_yday - 1) % 7;
          (A == 4 || A == 5 && Ee(l.tm_year % 400 - 1)) && v++;
        }
        return h(v, 2);
      }, "%w": function(l) {
        return l.tm_wday;
      }, "%W": function(l) {
        var v = l.tm_yday + 7 - (l.tm_wday + 6) % 7;
        return h(Math.floor(v / 7), 2);
      }, "%y": function(l) {
        return (l.tm_year + 1900).toString().substring(2);
      }, "%Y": function(l) {
        return l.tm_year + 1900;
      }, "%z": function(l) {
        var v = l.tm_gmtoff, A = v >= 0;
        return v = Math.abs(v) / 60, v = v / 60 * 100 + v % 60, (A ? "+" : "-") + ("0000" + v).slice(-4);
      }, "%Z": function(l) {
        return l.tm_zone;
      }, "%%": function() {
        return "%";
      } };
      s = s.replace(/%%/g, "\0\0");
      for (var f in C)
        s.includes(f) && (s = s.replace(new RegExp(f, "g"), C[f](a)));
      s = s.replace(/\0\0/g, "%");
      var P = Tn(s, !1);
      return P.length > r ? 0 : (An(P, e), P.length - 1);
    }
    function En(e, r, t, n, i) {
      return Fn(e, r, t, n);
    }
    sr = u.InternalError = Ue(Error, "InternalError"), ut(), re = u.BindingError = Ue(Error, "BindingError"), Pt(), _t(), Dt(), _r = u.UnboundTypeError = Ue(Error, "UnboundTypeError"), Vt();
    var Sn = { r: Zr, v: Gr, a: Jr, i: Kr, n: Qr, J: et, T: rt, p: tt, ca: nt, d: qr, na: ot, Z: st, ga: ft, X: Ut, ma: Ht, F: Mt, V: It, fa: Bt, S: Nt, O: zt, y: Zt, t: Gt, R: Kt, L: on, P: sn, oa: un, ha: fn, ka: mr, la: cn, W: dn, K: vn, ea: pn, da: gn, aa: wn, ba: bn, C: Gn, Q: ni, E: qn, q: zn, b: Rn, w: Zn, ia: Kn, c: Wn, h: Vn, j: Ln, l: Bn, z: Xn, x: Nn, I: ei, N: ri, D: Jn, H: ii, _: oi, Y: si, k: jn, g: On, f: Dn, e: kn, M: ti, m: xn, o: Un, ja: Mn, u: In, U: Yn, B: Qn, s: Hn, G: ai, A: $n, $: En };
    Xr();
    var V = u._free = function() {
      return (V = u._free = u.asm.ra).apply(null, arguments);
    }, ze = u._malloc = function() {
      return (ze = u._malloc = u.asm.sa).apply(null, arguments);
    }, Ar = function() {
      return (Ar = u.asm.ua).apply(null, arguments);
    };
    u.__embind_initialize_bindings = function() {
      return (u.__embind_initialize_bindings = u.asm.va).apply(null, arguments);
    };
    var y = function() {
      return (y = u.asm.wa).apply(null, arguments);
    }, de = function() {
      return (de = u.asm.xa).apply(null, arguments);
    }, m = function() {
      return (m = u.asm.ya).apply(null, arguments);
    }, w = function() {
      return (w = u.asm.za).apply(null, arguments);
    }, Fr = function() {
      return (Fr = u.asm.Aa).apply(null, arguments);
    }, Er = function() {
      return (Er = u.asm.Ba).apply(null, arguments);
    }, Sr = function() {
      return (Sr = u.asm.Ca).apply(null, arguments);
    }, Rr = function() {
      return (Rr = u.asm.Da).apply(null, arguments);
    };
    u.dynCall_viijii = function() {
      return (u.dynCall_viijii = u.asm.Ea).apply(null, arguments);
    };
    var kr = u.dynCall_jiii = function() {
      return (kr = u.dynCall_jiii = u.asm.Fa).apply(null, arguments);
    }, Dr = u.dynCall_jiiii = function() {
      return (Dr = u.dynCall_jiiii = u.asm.Ga).apply(null, arguments);
    };
    u.dynCall_iiiiij = function() {
      return (u.dynCall_iiiiij = u.asm.Ha).apply(null, arguments);
    }, u.dynCall_iiiiijj = function() {
      return (u.dynCall_iiiiijj = u.asm.Ia).apply(null, arguments);
    }, u.dynCall_iiiiiijj = function() {
      return (u.dynCall_iiiiiijj = u.asm.Ja).apply(null, arguments);
    };
    function Rn(e, r) {
      var t = m();
      try {
        return b(e)(r);
      } catch (n) {
        if (w(t), n !== n + 0)
          throw n;
        y(1, 0);
      }
    }
    function kn(e, r, t, n) {
      var i = m();
      try {
        b(e)(r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function Dn(e, r, t) {
      var n = m();
      try {
        b(e)(r, t);
      } catch (i) {
        if (w(n), i !== i + 0)
          throw i;
        y(1, 0);
      }
    }
    function On(e, r) {
      var t = m();
      try {
        b(e)(r);
      } catch (n) {
        if (w(t), n !== n + 0)
          throw n;
        y(1, 0);
      }
    }
    function Wn(e, r, t) {
      var n = m();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (w(n), i !== i + 0)
          throw i;
        y(1, 0);
      }
    }
    function jn(e) {
      var r = m();
      try {
        b(e)();
      } catch (t) {
        if (w(r), t !== t + 0)
          throw t;
        y(1, 0);
      }
    }
    function Un(e, r, t, n, i, a) {
      var s = m();
      try {
        b(e)(r, t, n, i, a);
      } catch (o) {
        if (w(s), o !== o + 0)
          throw o;
        y(1, 0);
      }
    }
    function Hn(e, r, t, n, i, a, s, o, f, c, d) {
      var p = m();
      try {
        b(e)(r, t, n, i, a, s, o, f, c, d);
      } catch (h) {
        if (w(p), h !== h + 0)
          throw h;
        y(1, 0);
      }
    }
    function Mn(e, r, t, n, i, a, s) {
      var o = m();
      try {
        b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (w(o), f !== f + 0)
          throw f;
        y(1, 0);
      }
    }
    function In(e, r, t, n, i, a, s, o) {
      var f = m();
      try {
        b(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (w(f), c !== c + 0)
          throw c;
        y(1, 0);
      }
    }
    function xn(e, r, t, n, i) {
      var a = m();
      try {
        b(e)(r, t, n, i);
      } catch (s) {
        if (w(a), s !== s + 0)
          throw s;
        y(1, 0);
      }
    }
    function Yn(e, r, t, n, i, a, s, o, f) {
      var c = m();
      try {
        b(e)(r, t, n, i, a, s, o, f);
      } catch (d) {
        if (w(c), d !== d + 0)
          throw d;
        y(1, 0);
      }
    }
    function Vn(e, r, t, n) {
      var i = m();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function Bn(e, r, t, n, i, a) {
      var s = m();
      try {
        return b(e)(r, t, n, i, a);
      } catch (o) {
        if (w(s), o !== o + 0)
          throw o;
        y(1, 0);
      }
    }
    function Ln(e, r, t, n, i) {
      var a = m();
      try {
        return b(e)(r, t, n, i);
      } catch (s) {
        if (w(a), s !== s + 0)
          throw s;
        y(1, 0);
      }
    }
    function Nn(e, r, t, n, i, a, s) {
      var o = m();
      try {
        return b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (w(o), f !== f + 0)
          throw f;
        y(1, 0);
      }
    }
    function zn(e) {
      var r = m();
      try {
        return b(e)();
      } catch (t) {
        if (w(r), t !== t + 0)
          throw t;
        y(1, 0);
      }
    }
    function Xn(e, r, t, n, i, a, s) {
      var o = m();
      try {
        return b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (w(o), f !== f + 0)
          throw f;
        y(1, 0);
      }
    }
    function Zn(e, r, t, n) {
      var i = m();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function Gn(e, r, t, n) {
      var i = m();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function qn(e, r, t, n, i, a) {
      var s = m();
      try {
        return b(e)(r, t, n, i, a);
      } catch (o) {
        if (w(s), o !== o + 0)
          throw o;
        y(1, 0);
      }
    }
    function Jn(e, r, t, n, i, a, s, o, f, c) {
      var d = m();
      try {
        return b(e)(r, t, n, i, a, s, o, f, c);
      } catch (p) {
        if (w(d), p !== p + 0)
          throw p;
        y(1, 0);
      }
    }
    function Kn(e, r, t) {
      var n = m();
      try {
        return b(e)(r, t);
      } catch (i) {
        if (w(n), i !== i + 0)
          throw i;
        y(1, 0);
      }
    }
    function Qn(e, r, t, n, i, a, s, o, f, c) {
      var d = m();
      try {
        b(e)(r, t, n, i, a, s, o, f, c);
      } catch (p) {
        if (w(d), p !== p + 0)
          throw p;
        y(1, 0);
      }
    }
    function ei(e, r, t, n, i, a, s, o) {
      var f = m();
      try {
        return b(e)(r, t, n, i, a, s, o);
      } catch (c) {
        if (w(f), c !== c + 0)
          throw c;
        y(1, 0);
      }
    }
    function ri(e, r, t, n, i, a, s, o, f) {
      var c = m();
      try {
        return b(e)(r, t, n, i, a, s, o, f);
      } catch (d) {
        if (w(c), d !== d + 0)
          throw d;
        y(1, 0);
      }
    }
    function ti(e, r, t, n, i, a, s) {
      var o = m();
      try {
        b(e)(r, t, n, i, a, s);
      } catch (f) {
        if (w(o), f !== f + 0)
          throw f;
        y(1, 0);
      }
    }
    function ni(e, r, t, n) {
      var i = m();
      try {
        return b(e)(r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function ii(e, r, t, n, i, a, s, o, f, c, d, p) {
      var h = m();
      try {
        return b(e)(r, t, n, i, a, s, o, f, c, d, p);
      } catch (_) {
        if (w(h), _ !== _ + 0)
          throw _;
        y(1, 0);
      }
    }
    function ai(e, r, t, n, i, a, s, o, f, c, d, p, h, _, $, T) {
      var C = m();
      try {
        b(e)(r, t, n, i, a, s, o, f, c, d, p, h, _, $, T);
      } catch (P) {
        if (w(C), P !== P + 0)
          throw P;
        y(1, 0);
      }
    }
    function oi(e, r, t, n) {
      var i = m();
      try {
        return kr(e, r, t, n);
      } catch (a) {
        if (w(i), a !== a + 0)
          throw a;
        y(1, 0);
      }
    }
    function si(e, r, t, n, i) {
      var a = m();
      try {
        return Dr(e, r, t, n, i);
      } catch (s) {
        if (w(a), s !== s + 0)
          throw s;
        y(1, 0);
      }
    }
    var Se;
    ae = function e() {
      Se || Or(), Se || (ae = e);
    };
    function Or() {
      if (z > 0 || (Ur(), z > 0))
        return;
      function e() {
        Se || (Se = !0, u.calledRun = !0, !Ze && (Hr(), B(u), u.onRuntimeInitialized && u.onRuntimeInitialized(), Mr()));
      }
      u.setStatus ? (u.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          u.setStatus("");
        }, 1), e();
      }, 1)) : e();
    }
    if (u.preInit)
      for (typeof u.preInit == "function" && (u.preInit = [u.preInit]); u.preInit.length > 0; )
        u.preInit.pop()();
    return Or(), H.ready;
  };
})();
function vi(W) {
  return ui(ke, W);
}
async function pi(W, {
  tryHarder: H = N.tryHarder,
  formats: u = N.formats,
  maxNumberOfSymbols: B = N.maxNumberOfSymbols
} = N) {
  return fi(
    W,
    {
      tryHarder: H,
      formats: u,
      maxNumberOfSymbols: B
    },
    ke
  );
}
async function hi(W, {
  tryHarder: H = N.tryHarder,
  formats: u = N.formats,
  maxNumberOfSymbols: B = N.maxNumberOfSymbols
} = N) {
  return ci(
    W,
    {
      tryHarder: H,
      formats: u,
      maxNumberOfSymbols: B
    },
    ke
  );
}
async function _i(W, {
  format: H = q.format,
  charset: u = q.charset,
  quietZone: B = q.quietZone,
  width: J = q.width,
  height: ve = q.height,
  eccLevel: pe = q.eccLevel
} = q) {
  return li(
    W,
    {
      format: H,
      charset: u,
      quietZone: B,
      width: J,
      height: ve,
      eccLevel: pe
    },
    ke
  );
}
export {
  mi as ZXING_BARCODE_FORMAT_NAMES,
  wi as ZXING_CHARACTOR_SET_NAMES,
  N as defaultZXingReadOptions,
  q as defaultZXingWriteOptions,
  vi as getZXingModule,
  bi as purgeZXingModule,
  hi as readBarcodeFromImageData,
  pi as readBarcodeFromImageFile,
  $i as setZXingModuleOverrides,
  _i as writeBarcodeToImageFile
};
