// @sec-ant/barcode-detector/dist/index.js
var jr = (f, m, i) => {
  if (!m.has(f))
    throw TypeError("Cannot " + i);
};
var Wr = (f, m, i) => (jr(f, m, "read from private field"), i ? i.call(f) : m.get(f)), Ir = (f, m, i) => {
  if (m.has(f))
    throw TypeError("Cannot add the same private member more than once");
  m instanceof WeakSet ? m.add(f) : m.set(f, i);
}, kr = (f, m, i, T) => (jr(f, m, "write to private field"), T ? T.call(f, i) : m.set(f, i), i);
function ga(f) {
  if (wa(f))
    return {
      width: f.naturalWidth,
      height: f.naturalHeight
    };
  if (ba(f))
    return {
      width: f.width.baseVal.value,
      height: f.height.baseVal.value
    };
  if ($a(f))
    return {
      width: f.videoWidth,
      height: f.videoHeight
    };
  if (Br(f))
    return {
      width: f.width,
      height: f.height
    };
  if (Ca(f))
    return {
      width: f.width,
      height: f.height
    };
  if (_a(f))
    return {
      width: f.width,
      height: f.height
    };
  if (Ta(f))
    return {
      width: f.displayWidth,
      height: f.displayHeight
    };
  throw new Error("Unsupported image type");
}
function wa(f) {
  try {
    return f instanceof HTMLImageElement;
  } catch {
    return !1;
  }
}
function ba(f) {
  try {
    return f instanceof SVGImageElement;
  } catch {
    return !1;
  }
}
function $a(f) {
  try {
    return f instanceof HTMLVideoElement;
  } catch {
    return !1;
  }
}
function Br(f) {
  try {
    return f instanceof HTMLCanvasElement;
  } catch {
    return !1;
  }
}
function Ca(f) {
  try {
    return f instanceof ImageBitmap;
  } catch {
    return !1;
  }
}
function _a(f) {
  try {
    return f instanceof OffscreenCanvas;
  } catch {
    return !1;
  }
}
function Ta(f) {
  try {
    return f instanceof VideoFrame;
  } catch {
    return !1;
  }
}
function Pa(f) {
  try {
    return f instanceof Blob;
  } catch {
    return !1;
  }
}
function Aa(f) {
  try {
    return f instanceof ImageData;
  } catch {
    return !1;
  }
}
function Ea(f, m) {
  try {
    const i = new OffscreenCanvas(f, m);
    if (i.getContext("2d") instanceof OffscreenCanvasRenderingContext2D)
      return i;
    throw new Error("OffscreenCanvasRenderingContext2D is not supported");
  } catch {
    const i = document.createElement("canvas");
    return i.width = f, i.height = m, i;
  }
}
function Vr(f) {
  const { width: m, height: i } = ga(f), T = Ea(m, i), P = T.getContext("2d");
  P.drawImage(f, 0, 0);
  const x = P.getImageData(0, 0, m, i);
  return Br(T) && T.remove(), x;
}
async function Sa(f) {
  const m = await createImageBitmap(f);
  return Vr(m);
}
async function xa(f) {
  return Pa(f) ? await Sa(f) : Aa(f) ? f : Vr(f);
}
const Hr = (f) => {
  let m;
  const i = /* @__PURE__ */ new Set(), T = (F, L) => {
    const I = typeof F == "function" ? F(m) : F;
    if (!Object.is(I, m)) {
      const M = m;
      m = L ?? typeof I != "object" ? I : Object.assign({}, m, I), i.forEach((V) => V(m, M));
    }
  }, P = () => m, x = { setState: T, getState: P, subscribe: (F) => (i.add(F), () => i.delete(F)), destroy: () => {
    ({ BASE_URL: "/", MODE: "production", DEV: !1, PROD: !0, SSR: !1 } && "production") !== "production" && console.warn(
      "[DEPRECATED] The `destroy` method will be unsupported in a future version. Instead use unsubscribe function returned by subscribe. Everything will be garbage-collected if store is garbage-collected."
    ), i.clear();
  } };
  return m = f(T, P, x), x;
}, Oa = (f) => f ? Hr(f) : Hr, Da = {
  locateFile: (f, m) => {
    var i;
    const T = (i = f.match(new RegExp("(?<=_).+?(?=\\.wasm$)"))) == null ? void 0 : i[0];
    return T ? `https://cdn.jsdelivr.net/npm/@sec-ant/zxing-wasm@1.2.4/dist/${T}/${f}` : m + f;
  }
}, ft = Oa()(() => ({
  zxingModuleWeakMap: /* @__PURE__ */ new WeakMap(),
  zxingModuleOverrides: Da
}));
function za(f) {
  ft.setState({
    zxingModuleOverrides: f
  });
}
function Yr(f, m = ft.getState().zxingModuleOverrides) {
  const { zxingModuleWeakMap: i } = ft.getState(), T = i.get(
    f
  );
  if (T && Object.is(m, ft.getState().zxingModuleOverrides))
    return T;
  {
    ft.setState({
      zxingModuleOverrides: m
    });
    const P = f(m);
    return i.set(f, P), P;
  }
}
const Ur = [
  "Aztec",
  "Codabar",
  "Code128",
  "Code39",
  "Code93",
  "DataBar",
  "DataBarExpanded",
  "DataMatrix",
  "EAN-13",
  "EAN-8",
  "ITF",
  "Linear-Codes",
  "Matrix-Codes",
  "MaxiCode",
  "MicroQRCode",
  "None",
  "PDF417",
  "QRCode",
  "UPC-A",
  "UPC-E"
], tt = {
  tryHarder: !0,
  formats: [],
  maxNumberOfSymbols: 1 / 0
};
async function Fa(f, {
  tryHarder: m = tt.tryHarder,
  formats: i = tt.formats,
  maxNumberOfSymbols: T = tt.maxNumberOfSymbols
} = tt, P) {
  const x = await Yr(
    P,
    ft.getState().zxingModuleOverrides
  ), {
    data: F,
    width: L,
    height: I,
    data: { byteLength: M }
  } = f, V = x._malloc(M);
  x.HEAP8.set(F, V);
  const N = x.readBarcodeFromPixmap(
    V,
    L,
    I,
    m,
    Ma(i),
    T
  );
  x._free(V);
  const U = [];
  for (let z = 0; z < N.size(); ++z) {
    const G = N.get(z);
    U.push({
      ...G,
      format: Ra(G.format)
    });
  }
  return U;
}
function Ma(f) {
  return f.join("|");
}
function Ra(f) {
  const m = zr(f);
  let i = 0, T = Ur.length - 1;
  for (; i <= T; ) {
    const P = Math.floor((i + T) / 2), x = Ur[P], F = zr(x);
    if (F === m)
      return x;
    F < m ? i = P + 1 : T = P - 1;
  }
  return "None";
}
function zr(f) {
  return f.toLowerCase().replace(/_-\[\]/g, "");
}
var Lr = (() => {
  var f = import.meta.url;
  return function(m = {}) {
    var i = typeof m < "u" ? m : {}, T, P;
    i.ready = new Promise((t, r) => {
      T = t, P = r;
    });
    var x = Object.assign({}, i), F = "./this.program", L = typeof window == "object", I = typeof importScripts == "function";
    typeof process == "object" && typeof process.versions == "object" && process.versions.node;
    var M = "";
    function V(t) {
      return i.locateFile ? i.locateFile(t, M) : M + t;
    }
    var N;
    (L || I) && (I ? M = self.location.href : typeof document < "u" && document.currentScript && (M = document.currentScript.src), f && (M = f), M.indexOf("blob:") !== 0 ? M = M.substr(0, M.replace(/[?#].*/, "").lastIndexOf("/") + 1) : M = "", I && (N = (t) => {
      var r = new XMLHttpRequest();
      return r.open("GET", t, !1), r.responseType = "arraybuffer", r.send(null), new Uint8Array(r.response);
    })), i.print || console.log.bind(console);
    var U = i.printErr || console.error.bind(console);
    Object.assign(i, x), x = null, i.arguments && i.arguments, i.thisProgram && (F = i.thisProgram), i.quit && i.quit;
    var z;
    i.wasmBinary && (z = i.wasmBinary), i.noExitRuntime, typeof WebAssembly != "object" && ut("no native wasm support detected");
    var G, bt = !1;
    function jt(t, r) {
      t || ut(r);
    }
    var X, H, it, $t, R, O, Jt, Qt;
    function Zt() {
      var t = G.buffer;
      i.HEAP8 = X = new Int8Array(t), i.HEAP16 = it = new Int16Array(t), i.HEAP32 = R = new Int32Array(t), i.HEAPU8 = H = new Uint8Array(t), i.HEAPU16 = $t = new Uint16Array(t), i.HEAPU32 = O = new Uint32Array(t), i.HEAPF32 = Jt = new Float32Array(t), i.HEAPF64 = Qt = new Float64Array(t);
    }
    var Kt, tr = [], rr = [], nr = [];
    function Gr() {
      if (i.preRun)
        for (typeof i.preRun == "function" && (i.preRun = [i.preRun]); i.preRun.length; )
          Jr(i.preRun.shift());
      Wt(tr);
    }
    function Xr() {
      Wt(rr);
    }
    function qr() {
      if (i.postRun)
        for (typeof i.postRun == "function" && (i.postRun = [i.postRun]); i.postRun.length; )
          Zr(i.postRun.shift());
      Wt(nr);
    }
    function Jr(t) {
      tr.unshift(t);
    }
    function Qr(t) {
      rr.unshift(t);
    }
    function Zr(t) {
      nr.unshift(t);
    }
    var rt = 0, lt = null;
    function Kr(t) {
      rt++, i.monitorRunDependencies && i.monitorRunDependencies(rt);
    }
    function tn(t) {
      if (rt--, i.monitorRunDependencies && i.monitorRunDependencies(rt), rt == 0 && lt) {
        var r = lt;
        lt = null, r();
      }
    }
    function ut(t) {
      i.onAbort && i.onAbort(t), t = "Aborted(" + t + ")", U(t), bt = !0, t += ". Build with -sASSERTIONS for more info.";
      var r = new WebAssembly.RuntimeError(t);
      throw P(r), r;
    }
    var rn = "data:application/octet-stream;base64,";
    function er(t) {
      return t.startsWith(rn);
    }
    var nt;
    i.locateFile ? (nt = "zxing_reader.wasm", er(nt) || (nt = V(nt))) : nt = new URL("/reader/zxing_reader.wasm", self.location).href;
    function ar(t) {
      try {
        if (t == nt && z)
          return new Uint8Array(z);
        if (N)
          return N(t);
        throw "both async and sync fetching of the wasm failed";
      } catch (r) {
        ut(r);
      }
    }
    function nn(t) {
      return !z && (L || I) && typeof fetch == "function" ? fetch(t, { credentials: "same-origin" }).then((r) => {
        if (!r.ok)
          throw "failed to load wasm binary file at '" + t + "'";
        return r.arrayBuffer();
      }).catch(() => ar(t)) : Promise.resolve().then(() => ar(t));
    }
    function or(t, r, n) {
      return nn(t).then((e) => WebAssembly.instantiate(e, r)).then((e) => e).then(n, (e) => {
        U("failed to asynchronously prepare wasm: " + e), ut(e);
      });
    }
    function en(t, r, n, e) {
      return !t && typeof WebAssembly.instantiateStreaming == "function" && !er(r) && typeof fetch == "function" ? fetch(r, { credentials: "same-origin" }).then((a) => {
        var o = WebAssembly.instantiateStreaming(a, n);
        return o.then(e, function(s) {
          return U("wasm streaming compile failed: " + s), U("falling back to ArrayBuffer instantiation"), or(r, n, e);
        });
      }) : or(r, n, e);
    }
    function an() {
      var t = { a: He };
      function r(e, a) {
        var o = e.exports;
        return i.asm = o, G = i.asm.oa, Zt(), Kt = i.asm.sa, Qr(i.asm.pa), tn(), o;
      }
      Kr();
      function n(e) {
        r(e.instance);
      }
      if (i.instantiateWasm)
        try {
          return i.instantiateWasm(t, r);
        } catch (e) {
          U("Module.instantiateWasm callback failed with error: " + e), P(e);
        }
      return en(z, nt, t, n).catch(P), {};
    }
    function Wt(t) {
      for (; t.length > 0; )
        t.shift()(i);
    }
    var Ct = [], _t = 0;
    function on(t) {
      var r = new Tt(t);
      return r.get_caught() || (r.set_caught(!0), _t--), r.set_rethrown(!1), Ct.push(r), xr(r.excPtr), r.get_exception_ptr();
    }
    var q = 0;
    function un() {
      g(0);
      var t = Ct.pop();
      Sr(t.excPtr), q = 0;
    }
    function Tt(t) {
      this.excPtr = t, this.ptr = t - 24, this.set_type = function(r) {
        O[this.ptr + 4 >> 2] = r;
      }, this.get_type = function() {
        return O[this.ptr + 4 >> 2];
      }, this.set_destructor = function(r) {
        O[this.ptr + 8 >> 2] = r;
      }, this.get_destructor = function() {
        return O[this.ptr + 8 >> 2];
      }, this.set_caught = function(r) {
        r = r ? 1 : 0, X[this.ptr + 12 >> 0] = r;
      }, this.get_caught = function() {
        return X[this.ptr + 12 >> 0] != 0;
      }, this.set_rethrown = function(r) {
        r = r ? 1 : 0, X[this.ptr + 13 >> 0] = r;
      }, this.get_rethrown = function() {
        return X[this.ptr + 13 >> 0] != 0;
      }, this.init = function(r, n) {
        this.set_adjusted_ptr(0), this.set_type(r), this.set_destructor(n);
      }, this.set_adjusted_ptr = function(r) {
        O[this.ptr + 16 >> 2] = r;
      }, this.get_adjusted_ptr = function() {
        return O[this.ptr + 16 >> 2];
      }, this.get_exception_ptr = function() {
        var r = Dr(this.get_type());
        if (r)
          return O[this.excPtr >> 2];
        var n = this.get_adjusted_ptr();
        return n !== 0 ? n : this.excPtr;
      };
    }
    function sn(t) {
      throw q || (q = t), q;
    }
    function It() {
      var t = q;
      if (!t)
        return gt(0), 0;
      var r = new Tt(t);
      r.set_adjusted_ptr(t);
      var n = r.get_type();
      if (!n)
        return gt(0), t;
      for (var e = 0; e < arguments.length; e++) {
        var a = arguments[e];
        if (a === 0 || a === n)
          break;
        var o = r.ptr + 16;
        if (Or(a, n, o))
          return gt(a), t;
      }
      return gt(n), t;
    }
    var cn = It, fn = It, ln = It;
    function hn(t) {
      var r = new Tt(t).get_exception_ptr();
      return r;
    }
    function pn() {
      var t = Ct.pop();
      t || ut("no exception to throw");
      var r = t.excPtr;
      throw t.get_rethrown() || (Ct.push(t), t.set_rethrown(!0), t.set_caught(!1), _t++), q = r, q;
    }
    function dn(t, r, n) {
      var e = new Tt(t);
      throw e.init(r, n), q = t, _t++, q;
    }
    function mn() {
      return _t;
    }
    var Pt = {};
    function ir(t) {
      for (; t.length; ) {
        var r = t.pop(), n = t.pop();
        n(r);
      }
    }
    function ht(t) {
      return this.fromWireType(R[t >> 2]);
    }
    var st = {}, et = {}, At = {}, yn = 48, vn = 57;
    function ur(t) {
      if (t === void 0)
        return "_unknown";
      t = t.replace(/[^a-zA-Z0-9_]/g, "$");
      var r = t.charCodeAt(0);
      return r >= yn && r <= vn ? `_${t}` : t;
    }
    function sr(t, r) {
      return t = ur(t), { [t]: function() {
        return r.apply(this, arguments);
      } }[t];
    }
    function kt(t, r) {
      var n = sr(r, function(e) {
        this.name = r, this.message = e;
        var a = new Error(e).stack;
        a !== void 0 && (this.stack = this.toString() + `
` + a.replace(/^Error(:[^\n]*)?\n/, ""));
      });
      return n.prototype = Object.create(t.prototype), n.prototype.constructor = n, n.prototype.toString = function() {
        return this.message === void 0 ? this.name : `${this.name}: ${this.message}`;
      }, n;
    }
    var cr = void 0;
    function Et(t) {
      throw new cr(t);
    }
    function at(t, r, n) {
      t.forEach(function(u) {
        At[u] = r;
      });
      function e(u) {
        var c = n(u);
        c.length !== t.length && Et("Mismatched type converter count");
        for (var l = 0; l < t.length; ++l)
          J(t[l], c[l]);
      }
      var a = new Array(r.length), o = [], s = 0;
      r.forEach((u, c) => {
        et.hasOwnProperty(u) ? a[c] = et[u] : (o.push(u), st.hasOwnProperty(u) || (st[u] = []), st[u].push(() => {
          a[c] = et[u], ++s, s === o.length && e(a);
        }));
      }), o.length === 0 && e(a);
    }
    function gn(t) {
      var r = Pt[t];
      delete Pt[t];
      var n = r.rawConstructor, e = r.rawDestructor, a = r.fields, o = a.map((s) => s.getterReturnType).concat(a.map((s) => s.setterArgumentType));
      at([t], o, (s) => {
        var u = {};
        return a.forEach((c, l) => {
          var p = c.fieldName, y = s[l], v = c.getter, $ = c.getterContext, A = s[l + a.length], j = c.setter, E = c.setterContext;
          u[p] = { read: (W) => y.fromWireType(v($, W)), write: (W, h) => {
            var d = [];
            j(E, W, A.toWireType(d, h)), ir(d);
          } };
        }), [{ name: r.name, fromWireType: function(c) {
          var l = {};
          for (var p in u)
            l[p] = u[p].read(c);
          return e(c), l;
        }, toWireType: function(c, l) {
          for (var p in u)
            if (!(p in l))
              throw new TypeError(`Missing field: "${p}"`);
          var y = n();
          for (p in u)
            u[p].write(y, l[p]);
          return c !== null && c.push(e, y), y;
        }, argPackAdvance: 8, readValueFromPointer: ht, destructorFunction: e }];
      });
    }
    function wn(t, r, n, e, a) {
    }
    function Ht(t) {
      switch (t) {
        case 1:
          return 0;
        case 2:
          return 1;
        case 4:
          return 2;
        case 8:
          return 3;
        default:
          throw new TypeError(`Unknown type size: ${t}`);
      }
    }
    function bn() {
      for (var t = new Array(256), r = 0; r < 256; ++r)
        t[r] = String.fromCharCode(r);
      fr = t;
    }
    var fr = void 0;
    function k(t) {
      for (var r = "", n = t; H[n]; )
        r += fr[H[n++]];
      return r;
    }
    var ct = void 0;
    function C(t) {
      throw new ct(t);
    }
    function J(t, r, n = {}) {
      if (!("argPackAdvance" in r))
        throw new TypeError("registerType registeredInstance requires argPackAdvance");
      var e = r.name;
      if (t || C(`type "${e}" must have a positive integer typeid pointer`), et.hasOwnProperty(t)) {
        if (n.ignoreDuplicateRegistrations)
          return;
        C(`Cannot register type '${e}' twice`);
      }
      if (et[t] = r, delete At[t], st.hasOwnProperty(t)) {
        var a = st[t];
        delete st[t], a.forEach((o) => o());
      }
    }
    function $n(t, r, n, e, a) {
      var o = Ht(n);
      r = k(r), J(t, { name: r, fromWireType: function(s) {
        return !!s;
      }, toWireType: function(s, u) {
        return u ? e : a;
      }, argPackAdvance: 8, readValueFromPointer: function(s) {
        var u;
        if (n === 1)
          u = X;
        else if (n === 2)
          u = it;
        else if (n === 4)
          u = R;
        else
          throw new TypeError("Unknown boolean type size: " + r);
        return this.fromWireType(u[s >> o]);
      }, destructorFunction: null });
    }
    function Cn(t) {
      if (!(this instanceof K) || !(t instanceof K))
        return !1;
      for (var r = this.$$.ptrType.registeredClass, n = this.$$.ptr, e = t.$$.ptrType.registeredClass, a = t.$$.ptr; r.baseClass; )
        n = r.upcast(n), r = r.baseClass;
      for (; e.baseClass; )
        a = e.upcast(a), e = e.baseClass;
      return r === e && n === a;
    }
    function _n(t) {
      return { count: t.count, deleteScheduled: t.deleteScheduled, preservePointerOnDelete: t.preservePointerOnDelete, ptr: t.ptr, ptrType: t.ptrType, smartPtr: t.smartPtr, smartPtrType: t.smartPtrType };
    }
    function Ut(t) {
      function r(n) {
        return n.$$.ptrType.registeredClass.name;
      }
      C(r(t) + " instance already deleted");
    }
    var zt = !1;
    function lr(t) {
    }
    function Tn(t) {
      t.smartPtr ? t.smartPtrType.rawDestructor(t.smartPtr) : t.ptrType.registeredClass.rawDestructor(t.ptr);
    }
    function hr(t) {
      t.count.value -= 1;
      var r = t.count.value === 0;
      r && Tn(t);
    }
    function pr(t, r, n) {
      if (r === n)
        return t;
      if (n.baseClass === void 0)
        return null;
      var e = pr(t, r, n.baseClass);
      return e === null ? null : n.downcast(e);
    }
    var dr = {};
    function Pn() {
      return Object.keys(mt).length;
    }
    function An() {
      var t = [];
      for (var r in mt)
        mt.hasOwnProperty(r) && t.push(mt[r]);
      return t;
    }
    var pt = [];
    function Bt() {
      for (; pt.length; ) {
        var t = pt.pop();
        t.$$.deleteScheduled = !1, t.delete();
      }
    }
    var dt = void 0;
    function En(t) {
      dt = t, pt.length && dt && dt(Bt);
    }
    function Sn() {
      i.getInheritedInstanceCount = Pn, i.getLiveInheritedInstances = An, i.flushPendingDeletes = Bt, i.setDelayFunction = En;
    }
    var mt = {};
    function xn(t, r) {
      for (r === void 0 && C("ptr should not be undefined"); t.baseClass; )
        r = t.upcast(r), t = t.baseClass;
      return r;
    }
    function On(t, r) {
      return r = xn(t, r), mt[r];
    }
    function St(t, r) {
      (!r.ptrType || !r.ptr) && Et("makeClassHandle requires ptr and ptrType");
      var n = !!r.smartPtrType, e = !!r.smartPtr;
      return n !== e && Et("Both smartPtrType and smartPtr must be specified"), r.count = { value: 1 }, yt(Object.create(t, { $$: { value: r } }));
    }
    function Dn(t) {
      var r = this.getPointee(t);
      if (!r)
        return this.destructor(t), null;
      var n = On(this.registeredClass, r);
      if (n !== void 0) {
        if (n.$$.count.value === 0)
          return n.$$.ptr = r, n.$$.smartPtr = t, n.clone();
        var e = n.clone();
        return this.destructor(t), e;
      }
      function a() {
        return this.isSmartPointer ? St(this.registeredClass.instancePrototype, { ptrType: this.pointeeType, ptr: r, smartPtrType: this, smartPtr: t }) : St(this.registeredClass.instancePrototype, { ptrType: this, ptr: t });
      }
      var o = this.registeredClass.getActualType(r), s = dr[o];
      if (!s)
        return a.call(this);
      var u;
      this.isConst ? u = s.constPointerType : u = s.pointerType;
      var c = pr(r, this.registeredClass, u.registeredClass);
      return c === null ? a.call(this) : this.isSmartPointer ? St(u.registeredClass.instancePrototype, { ptrType: u, ptr: c, smartPtrType: this, smartPtr: t }) : St(u.registeredClass.instancePrototype, { ptrType: u, ptr: c });
    }
    function yt(t) {
      return typeof FinalizationRegistry > "u" ? (yt = (r) => r, t) : (zt = new FinalizationRegistry((r) => {
        hr(r.$$);
      }), yt = (r) => {
        var n = r.$$, e = !!n.smartPtr;
        if (e) {
          var a = { $$: n };
          zt.register(r, a, r);
        }
        return r;
      }, lr = (r) => zt.unregister(r), yt(t));
    }
    function Fn() {
      if (this.$$.ptr || Ut(this), this.$$.preservePointerOnDelete)
        return this.$$.count.value += 1, this;
      var t = yt(Object.create(Object.getPrototypeOf(this), { $$: { value: _n(this.$$) } }));
      return t.$$.count.value += 1, t.$$.deleteScheduled = !1, t;
    }
    function Mn() {
      this.$$.ptr || Ut(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && C("Object already scheduled for deletion"), lr(this), hr(this.$$), this.$$.preservePointerOnDelete || (this.$$.smartPtr = void 0, this.$$.ptr = void 0);
    }
    function Rn() {
      return !this.$$.ptr;
    }
    function jn() {
      return this.$$.ptr || Ut(this), this.$$.deleteScheduled && !this.$$.preservePointerOnDelete && C("Object already scheduled for deletion"), pt.push(this), pt.length === 1 && dt && dt(Bt), this.$$.deleteScheduled = !0, this;
    }
    function Wn() {
      K.prototype.isAliasOf = Cn, K.prototype.clone = Fn, K.prototype.delete = Mn, K.prototype.isDeleted = Rn, K.prototype.deleteLater = jn;
    }
    function K() {
    }
    function mr(t, r, n) {
      if (t[r].overloadTable === void 0) {
        var e = t[r];
        t[r] = function() {
          return t[r].overloadTable.hasOwnProperty(arguments.length) || C(`Function '${n}' called with an invalid number of arguments (${arguments.length}) - expects one of (${t[r].overloadTable})!`), t[r].overloadTable[arguments.length].apply(this, arguments);
        }, t[r].overloadTable = [], t[r].overloadTable[e.argCount] = e;
      }
    }
    function yr(t, r, n) {
      i.hasOwnProperty(t) ? ((n === void 0 || i[t].overloadTable !== void 0 && i[t].overloadTable[n] !== void 0) && C(`Cannot register public name '${t}' twice`), mr(i, t, t), i.hasOwnProperty(n) && C(`Cannot register multiple overloads of a function with the same number of arguments (${n})!`), i[t].overloadTable[n] = r) : (i[t] = r, n !== void 0 && (i[t].numArguments = n));
    }
    function In(t, r, n, e, a, o, s, u) {
      this.name = t, this.constructor = r, this.instancePrototype = n, this.rawDestructor = e, this.baseClass = a, this.getActualType = o, this.upcast = s, this.downcast = u, this.pureVirtualFunctions = [];
    }
    function Vt(t, r, n) {
      for (; r !== n; )
        r.upcast || C(`Expected null or instance of ${n.name}, got an instance of ${r.name}`), t = r.upcast(t), r = r.baseClass;
      return t;
    }
    function kn(t, r) {
      if (r === null)
        return this.isReference && C(`null is not a valid ${this.name}`), 0;
      r.$$ || C(`Cannot pass "${Nt(r)}" as a ${this.name}`), r.$$.ptr || C(`Cannot pass deleted object as a pointer of type ${this.name}`);
      var n = r.$$.ptrType.registeredClass, e = Vt(r.$$.ptr, n, this.registeredClass);
      return e;
    }
    function Hn(t, r) {
      var n;
      if (r === null)
        return this.isReference && C(`null is not a valid ${this.name}`), this.isSmartPointer ? (n = this.rawConstructor(), t !== null && t.push(this.rawDestructor, n), n) : 0;
      r.$$ || C(`Cannot pass "${Nt(r)}" as a ${this.name}`), r.$$.ptr || C(`Cannot pass deleted object as a pointer of type ${this.name}`), !this.isConst && r.$$.ptrType.isConst && C(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
      var e = r.$$.ptrType.registeredClass;
      if (n = Vt(r.$$.ptr, e, this.registeredClass), this.isSmartPointer)
        switch (r.$$.smartPtr === void 0 && C("Passing raw pointer to smart pointer is illegal"), this.sharingPolicy) {
          case 0:
            r.$$.smartPtrType === this ? n = r.$$.smartPtr : C(`Cannot convert argument of type ${r.$$.smartPtrType ? r.$$.smartPtrType.name : r.$$.ptrType.name} to parameter type ${this.name}`);
            break;
          case 1:
            n = r.$$.smartPtr;
            break;
          case 2:
            if (r.$$.smartPtrType === this)
              n = r.$$.smartPtr;
            else {
              var a = r.clone();
              n = this.rawShare(n, Dt.toHandle(function() {
                a.delete();
              })), t !== null && t.push(this.rawDestructor, n);
            }
            break;
          default:
            C("Unsupporting sharing policy");
        }
      return n;
    }
    function Un(t, r) {
      if (r === null)
        return this.isReference && C(`null is not a valid ${this.name}`), 0;
      r.$$ || C(`Cannot pass "${Nt(r)}" as a ${this.name}`), r.$$.ptr || C(`Cannot pass deleted object as a pointer of type ${this.name}`), r.$$.ptrType.isConst && C(`Cannot convert argument of type ${r.$$.ptrType.name} to parameter type ${this.name}`);
      var n = r.$$.ptrType.registeredClass, e = Vt(r.$$.ptr, n, this.registeredClass);
      return e;
    }
    function zn(t) {
      return this.rawGetPointee && (t = this.rawGetPointee(t)), t;
    }
    function Bn(t) {
      this.rawDestructor && this.rawDestructor(t);
    }
    function Vn(t) {
      t !== null && t.delete();
    }
    function Yn() {
      Q.prototype.getPointee = zn, Q.prototype.destructor = Bn, Q.prototype.argPackAdvance = 8, Q.prototype.readValueFromPointer = ht, Q.prototype.deleteObject = Vn, Q.prototype.fromWireType = Dn;
    }
    function Q(t, r, n, e, a, o, s, u, c, l, p) {
      this.name = t, this.registeredClass = r, this.isReference = n, this.isConst = e, this.isSmartPointer = a, this.pointeeType = o, this.sharingPolicy = s, this.rawGetPointee = u, this.rawConstructor = c, this.rawShare = l, this.rawDestructor = p, !a && r.baseClass === void 0 ? e ? (this.toWireType = kn, this.destructorFunction = null) : (this.toWireType = Un, this.destructorFunction = null) : this.toWireType = Hn;
    }
    function vr(t, r, n) {
      i.hasOwnProperty(t) || Et("Replacing nonexistant public symbol"), i[t].overloadTable !== void 0 && n !== void 0 ? i[t].overloadTable[n] = r : (i[t] = r, i[t].argCount = n);
    }
    function Ln(t, r, n) {
      var e = i["dynCall_" + t];
      return n && n.length ? e.apply(null, [r].concat(n)) : e.call(null, r);
    }
    var xt = [];
    function _(t) {
      var r = xt[t];
      return r || (t >= xt.length && (xt.length = t + 1), xt[t] = r = Kt.get(t)), r;
    }
    function Nn(t, r, n) {
      if (t.includes("j"))
        return Ln(t, r, n);
      var e = _(r).apply(null, n);
      return e;
    }
    function Gn(t, r) {
      var n = [];
      return function() {
        return n.length = 0, Object.assign(n, arguments), Nn(t, r, n);
      };
    }
    function Y(t, r) {
      t = k(t);
      function n() {
        return t.includes("j") ? Gn(t, r) : _(r);
      }
      var e = n();
      return typeof e != "function" && C(`unknown function pointer with signature ${t}: ${r}`), e;
    }
    var gr = void 0;
    function wr(t) {
      var r = Er(t), n = k(r);
      return Z(r), n;
    }
    function Ot(t, r) {
      var n = [], e = {};
      function a(o) {
        if (!e[o] && !et[o]) {
          if (At[o]) {
            At[o].forEach(a);
            return;
          }
          n.push(o), e[o] = !0;
        }
      }
      throw r.forEach(a), new gr(`${t}: ` + n.map(wr).join([", "]));
    }
    function Xn(t, r, n, e, a, o, s, u, c, l, p, y, v) {
      p = k(p), o = Y(a, o), u && (u = Y(s, u)), l && (l = Y(c, l)), v = Y(y, v);
      var $ = ur(p);
      yr($, function() {
        Ot(`Cannot construct ${p} due to unbound types`, [e]);
      }), at([t, r, n], e ? [e] : [], function(A) {
        A = A[0];
        var j, E;
        e ? (j = A.registeredClass, E = j.instancePrototype) : E = K.prototype;
        var W = sr($, function() {
          if (Object.getPrototypeOf(this) !== h)
            throw new ct("Use 'new' to construct " + p);
          if (d.constructor_body === void 0)
            throw new ct(p + " has no accessible constructor");
          var Rt = d.constructor_body[arguments.length];
          if (Rt === void 0)
            throw new ct(`Tried to invoke ctor of ${p} with invalid number of parameters (${arguments.length}) - expected (${Object.keys(d.constructor_body).toString()}) parameters instead!`);
          return Rt.apply(this, arguments);
        }), h = Object.create(E, { constructor: { value: W } });
        W.prototype = h;
        var d = new In(p, W, h, v, j, o, u, l);
        d.baseClass && (d.baseClass.__derivedClasses === void 0 && (d.baseClass.__derivedClasses = []), d.baseClass.__derivedClasses.push(d));
        var S = new Q(p, d, !0, !1, !1), D = new Q(p + "*", d, !1, !1, !1), ot = new Q(p + " const*", d, !1, !0, !1);
        return dr[t] = { pointerType: D, constPointerType: ot }, vr($, W), [S, D, ot];
      });
    }
    function Yt(t, r) {
      for (var n = [], e = 0; e < t; e++)
        n.push(O[r + e * 4 >> 2]);
      return n;
    }
    function Lt(t, r, n, e, a, o) {
      var s = r.length;
      s < 2 && C("argTypes array size mismatch! Must at least get return value and 'this' types!");
      for (var u = r[1] !== null && n !== null, c = !1, l = 1; l < r.length; ++l)
        if (r[l] !== null && r[l].destructorFunction === void 0) {
          c = !0;
          break;
        }
      var p = r[0].name !== "void", y = s - 2, v = new Array(y), $ = [], A = [];
      return function() {
        arguments.length !== y && C(`function ${t} called with ${arguments.length} arguments, expected ${y} args!`), A.length = 0;
        var j;
        $.length = u ? 2 : 1, $[0] = a, u && (j = r[1].toWireType(A, this), $[1] = j);
        for (var E = 0; E < y; ++E)
          v[E] = r[E + 2].toWireType(A, arguments[E]), $.push(v[E]);
        var W = e.apply(null, $);
        function h(d) {
          if (c)
            ir(A);
          else
            for (var S = u ? 1 : 2; S < r.length; S++) {
              var D = S === 1 ? j : v[S - 2];
              r[S].destructorFunction !== null && r[S].destructorFunction(D);
            }
          if (p)
            return r[0].fromWireType(d);
        }
        return h(W);
      };
    }
    function qn(t, r, n, e, a, o) {
      jt(r > 0);
      var s = Yt(r, n);
      a = Y(e, a), at([], [t], function(u) {
        u = u[0];
        var c = `constructor ${u.name}`;
        if (u.registeredClass.constructor_body === void 0 && (u.registeredClass.constructor_body = []), u.registeredClass.constructor_body[r - 1] !== void 0)
          throw new ct(`Cannot register multiple constructors with identical number of parameters (${r - 1}) for class '${u.name}'! Overload resolution is currently only performed using the parameter count, not actual type info!`);
        return u.registeredClass.constructor_body[r - 1] = () => {
          Ot(`Cannot construct ${u.name} due to unbound types`, s);
        }, at([], s, function(l) {
          return l.splice(1, 0, null), u.registeredClass.constructor_body[r - 1] = Lt(c, l, null, a, o), [];
        }), [];
      });
    }
    function Jn(t, r, n, e, a, o, s, u, c) {
      var l = Yt(n, e);
      r = k(r), o = Y(a, o), at([], [t], function(p) {
        p = p[0];
        var y = `${p.name}.${r}`;
        r.startsWith("@@") && (r = Symbol[r.substring(2)]), u && p.registeredClass.pureVirtualFunctions.push(r);
        function v() {
          Ot(`Cannot call ${y} due to unbound types`, l);
        }
        var $ = p.registeredClass.instancePrototype, A = $[r];
        return A === void 0 || A.overloadTable === void 0 && A.className !== p.name && A.argCount === n - 2 ? (v.argCount = n - 2, v.className = p.name, $[r] = v) : (mr($, r, y), $[r].overloadTable[n - 2] = v), at([], l, function(j) {
          var E = Lt(y, j, p, o, s);
          return $[r].overloadTable === void 0 ? (E.argCount = n - 2, $[r] = E) : $[r].overloadTable[n - 2] = E, [];
        }), [];
      });
    }
    function Qn() {
      this.allocated = [void 0], this.freelist = [], this.get = function(t) {
        return this.allocated[t];
      }, this.has = function(t) {
        return this.allocated[t] !== void 0;
      }, this.allocate = function(t) {
        var r = this.freelist.pop() || this.allocated.length;
        return this.allocated[r] = t, r;
      }, this.free = function(t) {
        this.allocated[t] = void 0, this.freelist.push(t);
      };
    }
    var B = new Qn();
    function br(t) {
      t >= B.reserved && --B.get(t).refcount === 0 && B.free(t);
    }
    function Zn() {
      for (var t = 0, r = B.reserved; r < B.allocated.length; ++r)
        B.allocated[r] !== void 0 && ++t;
      return t;
    }
    function Kn() {
      B.allocated.push({ value: void 0 }, { value: null }, { value: !0 }, { value: !1 }), B.reserved = B.allocated.length, i.count_emval_handles = Zn;
    }
    var Dt = { toValue: (t) => (t || C("Cannot use deleted val. handle = " + t), B.get(t).value), toHandle: (t) => {
      switch (t) {
        case void 0:
          return 1;
        case null:
          return 2;
        case !0:
          return 3;
        case !1:
          return 4;
        default:
          return B.allocate({ refcount: 1, value: t });
      }
    } };
    function te(t, r) {
      r = k(r), J(t, { name: r, fromWireType: function(n) {
        var e = Dt.toValue(n);
        return br(n), e;
      }, toWireType: function(n, e) {
        return Dt.toHandle(e);
      }, argPackAdvance: 8, readValueFromPointer: ht, destructorFunction: null });
    }
    function Nt(t) {
      if (t === null)
        return "null";
      var r = typeof t;
      return r === "object" || r === "array" || r === "function" ? t.toString() : "" + t;
    }
    function re(t, r) {
      switch (r) {
        case 2:
          return function(n) {
            return this.fromWireType(Jt[n >> 2]);
          };
        case 3:
          return function(n) {
            return this.fromWireType(Qt[n >> 3]);
          };
        default:
          throw new TypeError("Unknown float type: " + t);
      }
    }
    function ne(t, r, n) {
      var e = Ht(n);
      r = k(r), J(t, { name: r, fromWireType: function(a) {
        return a;
      }, toWireType: function(a, o) {
        return o;
      }, argPackAdvance: 8, readValueFromPointer: re(r, e), destructorFunction: null });
    }
    function ee(t, r, n, e, a, o, s) {
      var u = Yt(r, n);
      t = k(t), a = Y(e, a), yr(t, function() {
        Ot(`Cannot call ${t} due to unbound types`, u);
      }, r - 1), at([], u, function(c) {
        var l = [c[0], null].concat(c.slice(1));
        return vr(t, Lt(t, l, null, a, o), r - 1), [];
      });
    }
    function ae(t, r, n) {
      switch (r) {
        case 0:
          return n ? function(e) {
            return X[e];
          } : function(e) {
            return H[e];
          };
        case 1:
          return n ? function(e) {
            return it[e >> 1];
          } : function(e) {
            return $t[e >> 1];
          };
        case 2:
          return n ? function(e) {
            return R[e >> 2];
          } : function(e) {
            return O[e >> 2];
          };
        default:
          throw new TypeError("Unknown integer type: " + t);
      }
    }
    function oe(t, r, n, e, a) {
      r = k(r);
      var o = Ht(n), s = (y) => y;
      if (e === 0) {
        var u = 32 - 8 * n;
        s = (y) => y << u >>> u;
      }
      var c = r.includes("unsigned"), l = (y, v) => {
      }, p;
      c ? p = function(y, v) {
        return l(v, this.name), v >>> 0;
      } : p = function(y, v) {
        return l(v, this.name), v;
      }, J(t, { name: r, fromWireType: s, toWireType: p, argPackAdvance: 8, readValueFromPointer: ae(r, o, e !== 0), destructorFunction: null });
    }
    function ie(t, r, n) {
      var e = [Int8Array, Uint8Array, Int16Array, Uint16Array, Int32Array, Uint32Array, Float32Array, Float64Array], a = e[r];
      function o(s) {
        s = s >> 2;
        var u = O, c = u[s], l = u[s + 1];
        return new a(u.buffer, l, c);
      }
      n = k(n), J(t, { name: n, fromWireType: o, argPackAdvance: 8, readValueFromPointer: o }, { ignoreDuplicateRegistrations: !0 });
    }
    function $r(t, r, n, e) {
      if (!(e > 0))
        return 0;
      for (var a = n, o = n + e - 1, s = 0; s < t.length; ++s) {
        var u = t.charCodeAt(s);
        if (u >= 55296 && u <= 57343) {
          var c = t.charCodeAt(++s);
          u = 65536 + ((u & 1023) << 10) | c & 1023;
        }
        if (u <= 127) {
          if (n >= o)
            break;
          r[n++] = u;
        } else if (u <= 2047) {
          if (n + 1 >= o)
            break;
          r[n++] = 192 | u >> 6, r[n++] = 128 | u & 63;
        } else if (u <= 65535) {
          if (n + 2 >= o)
            break;
          r[n++] = 224 | u >> 12, r[n++] = 128 | u >> 6 & 63, r[n++] = 128 | u & 63;
        } else {
          if (n + 3 >= o)
            break;
          r[n++] = 240 | u >> 18, r[n++] = 128 | u >> 12 & 63, r[n++] = 128 | u >> 6 & 63, r[n++] = 128 | u & 63;
        }
      }
      return r[n] = 0, n - a;
    }
    function ue(t, r, n) {
      return $r(t, H, r, n);
    }
    function Cr(t) {
      for (var r = 0, n = 0; n < t.length; ++n) {
        var e = t.charCodeAt(n);
        e <= 127 ? r++ : e <= 2047 ? r += 2 : e >= 55296 && e <= 57343 ? (r += 4, ++n) : r += 3;
      }
      return r;
    }
    var _r = typeof TextDecoder < "u" ? new TextDecoder("utf8") : void 0;
    function se(t, r, n) {
      for (var e = r + n, a = r; t[a] && !(a >= e); )
        ++a;
      if (a - r > 16 && t.buffer && _r)
        return _r.decode(t.subarray(r, a));
      for (var o = ""; r < a; ) {
        var s = t[r++];
        if (!(s & 128)) {
          o += String.fromCharCode(s);
          continue;
        }
        var u = t[r++] & 63;
        if ((s & 224) == 192) {
          o += String.fromCharCode((s & 31) << 6 | u);
          continue;
        }
        var c = t[r++] & 63;
        if ((s & 240) == 224 ? s = (s & 15) << 12 | u << 6 | c : s = (s & 7) << 18 | u << 12 | c << 6 | t[r++] & 63, s < 65536)
          o += String.fromCharCode(s);
        else {
          var l = s - 65536;
          o += String.fromCharCode(55296 | l >> 10, 56320 | l & 1023);
        }
      }
      return o;
    }
    function Gt(t, r) {
      return t ? se(H, t, r) : "";
    }
    function ce(t, r) {
      r = k(r);
      var n = r === "std::string";
      J(t, { name: r, fromWireType: function(e) {
        var a = O[e >> 2], o = e + 4, s;
        if (n)
          for (var u = o, c = 0; c <= a; ++c) {
            var l = o + c;
            if (c == a || H[l] == 0) {
              var p = l - u, y = Gt(u, p);
              s === void 0 ? s = y : (s += String.fromCharCode(0), s += y), u = l + 1;
            }
          }
        else {
          for (var v = new Array(a), c = 0; c < a; ++c)
            v[c] = String.fromCharCode(H[o + c]);
          s = v.join("");
        }
        return Z(e), s;
      }, toWireType: function(e, a) {
        a instanceof ArrayBuffer && (a = new Uint8Array(a));
        var o, s = typeof a == "string";
        s || a instanceof Uint8Array || a instanceof Uint8ClampedArray || a instanceof Int8Array || C("Cannot pass non-string to std::string"), n && s ? o = Cr(a) : o = a.length;
        var u = qt(4 + o + 1), c = u + 4;
        if (O[u >> 2] = o, n && s)
          ue(a, c, o + 1);
        else if (s)
          for (var l = 0; l < o; ++l) {
            var p = a.charCodeAt(l);
            p > 255 && (Z(c), C("String has UTF-16 code units that do not fit in 8 bits")), H[c + l] = p;
          }
        else
          for (var l = 0; l < o; ++l)
            H[c + l] = a[l];
        return e !== null && e.push(Z, u), u;
      }, argPackAdvance: 8, readValueFromPointer: ht, destructorFunction: function(e) {
        Z(e);
      } });
    }
    var Tr = typeof TextDecoder < "u" ? new TextDecoder("utf-16le") : void 0;
    function fe(t, r) {
      for (var n = t, e = n >> 1, a = e + r / 2; !(e >= a) && $t[e]; )
        ++e;
      if (n = e << 1, n - t > 32 && Tr)
        return Tr.decode(H.subarray(t, n));
      for (var o = "", s = 0; !(s >= r / 2); ++s) {
        var u = it[t + s * 2 >> 1];
        if (u == 0)
          break;
        o += String.fromCharCode(u);
      }
      return o;
    }
    function le(t, r, n) {
      if (n === void 0 && (n = 2147483647), n < 2)
        return 0;
      n -= 2;
      for (var e = r, a = n < t.length * 2 ? n / 2 : t.length, o = 0; o < a; ++o) {
        var s = t.charCodeAt(o);
        it[r >> 1] = s, r += 2;
      }
      return it[r >> 1] = 0, r - e;
    }
    function he(t) {
      return t.length * 2;
    }
    function pe(t, r) {
      for (var n = 0, e = ""; !(n >= r / 4); ) {
        var a = R[t + n * 4 >> 2];
        if (a == 0)
          break;
        if (++n, a >= 65536) {
          var o = a - 65536;
          e += String.fromCharCode(55296 | o >> 10, 56320 | o & 1023);
        } else
          e += String.fromCharCode(a);
      }
      return e;
    }
    function de(t, r, n) {
      if (n === void 0 && (n = 2147483647), n < 4)
        return 0;
      for (var e = r, a = e + n - 4, o = 0; o < t.length; ++o) {
        var s = t.charCodeAt(o);
        if (s >= 55296 && s <= 57343) {
          var u = t.charCodeAt(++o);
          s = 65536 + ((s & 1023) << 10) | u & 1023;
        }
        if (R[r >> 2] = s, r += 4, r + 4 > a)
          break;
      }
      return R[r >> 2] = 0, r - e;
    }
    function me(t) {
      for (var r = 0, n = 0; n < t.length; ++n) {
        var e = t.charCodeAt(n);
        e >= 55296 && e <= 57343 && ++n, r += 4;
      }
      return r;
    }
    function ye(t, r, n) {
      n = k(n);
      var e, a, o, s, u;
      r === 2 ? (e = fe, a = le, s = he, o = () => $t, u = 1) : r === 4 && (e = pe, a = de, s = me, o = () => O, u = 2), J(t, { name: n, fromWireType: function(c) {
        for (var l = O[c >> 2], p = o(), y, v = c + 4, $ = 0; $ <= l; ++$) {
          var A = c + 4 + $ * r;
          if ($ == l || p[A >> u] == 0) {
            var j = A - v, E = e(v, j);
            y === void 0 ? y = E : (y += String.fromCharCode(0), y += E), v = A + r;
          }
        }
        return Z(c), y;
      }, toWireType: function(c, l) {
        typeof l != "string" && C(`Cannot pass non-string to C++ string type ${n}`);
        var p = s(l), y = qt(4 + p + r);
        return O[y >> 2] = p >> u, a(l, y + 4, p + r), c !== null && c.push(Z, y), y;
      }, argPackAdvance: 8, readValueFromPointer: ht, destructorFunction: function(c) {
        Z(c);
      } });
    }
    function ve(t, r, n, e, a, o) {
      Pt[t] = { name: k(r), rawConstructor: Y(n, e), rawDestructor: Y(a, o), fields: [] };
    }
    function ge(t, r, n, e, a, o, s, u, c, l) {
      Pt[t].fields.push({ fieldName: k(r), getterReturnType: n, getter: Y(e, a), getterContext: o, setterArgumentType: s, setter: Y(u, c), setterContext: l });
    }
    function we(t, r) {
      r = k(r), J(t, { isVoid: !0, name: r, argPackAdvance: 0, fromWireType: function() {
      }, toWireType: function(n, e) {
      } });
    }
    function be(t) {
      t > 4 && (B.get(t).refcount += 1);
    }
    function $e(t, r) {
      var n = et[t];
      return n === void 0 && C(r + " has unknown type " + wr(t)), n;
    }
    function Ce(t, r) {
      t = $e(t, "_emval_take_value");
      var n = t.readValueFromPointer(r);
      return Dt.toHandle(n);
    }
    function _e() {
      ut("");
    }
    function Te(t, r, n) {
      H.copyWithin(t, r, r + n);
    }
    function Pe() {
      return 2147483648;
    }
    function Ae(t) {
      var r = G.buffer, n = t - r.byteLength + 65535 >>> 16;
      try {
        return G.grow(n), Zt(), 1;
      } catch {
      }
    }
    function Ee(t) {
      var r = H.length;
      t = t >>> 0;
      var n = Pe();
      if (t > n)
        return !1;
      for (var e = (c, l) => c + (l - c % l) % l, a = 1; a <= 4; a *= 2) {
        var o = r * (1 + 0.2 / a);
        o = Math.min(o, t + 100663296);
        var s = Math.min(n, e(Math.max(t, o), 65536)), u = Ae(s);
        if (u)
          return !0;
      }
      return !1;
    }
    var Xt = {};
    function Se() {
      return F || "./this.program";
    }
    function vt() {
      if (!vt.strings) {
        var t = (typeof navigator == "object" && navigator.languages && navigator.languages[0] || "C").replace("-", "_") + ".UTF-8", r = { USER: "web_user", LOGNAME: "web_user", PATH: "/", PWD: "/", HOME: "/home/web_user", LANG: t, _: Se() };
        for (var n in Xt)
          Xt[n] === void 0 ? delete r[n] : r[n] = Xt[n];
        var e = [];
        for (var n in r)
          e.push(`${n}=${r[n]}`);
        vt.strings = e;
      }
      return vt.strings;
    }
    function xe(t, r) {
      for (var n = 0; n < t.length; ++n)
        X[r++ >> 0] = t.charCodeAt(n);
      X[r >> 0] = 0;
    }
    function Oe(t, r) {
      var n = 0;
      return vt().forEach(function(e, a) {
        var o = r + n;
        O[t + a * 4 >> 2] = o, xe(e, o), n += e.length + 1;
      }), 0;
    }
    function De(t, r) {
      var n = vt();
      O[t >> 2] = n.length;
      var e = 0;
      return n.forEach(function(a) {
        e += a.length + 1;
      }), O[r >> 2] = e, 0;
    }
    function Fe(t) {
      return t;
    }
    function Ft(t) {
      return t % 4 === 0 && (t % 100 !== 0 || t % 400 === 0);
    }
    function Me(t, r) {
      for (var n = 0, e = 0; e <= r; n += t[e++])
        ;
      return n;
    }
    var Pr = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31], Ar = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    function Re(t, r) {
      for (var n = new Date(t.getTime()); r > 0; ) {
        var e = Ft(n.getFullYear()), a = n.getMonth(), o = (e ? Pr : Ar)[a];
        if (r > o - n.getDate())
          r -= o - n.getDate() + 1, n.setDate(1), a < 11 ? n.setMonth(a + 1) : (n.setMonth(0), n.setFullYear(n.getFullYear() + 1));
        else
          return n.setDate(n.getDate() + r), n;
      }
      return n;
    }
    function je(t, r, n) {
      var e = n > 0 ? n : Cr(t) + 1, a = new Array(e), o = $r(t, a, 0, a.length);
      return r && (a.length = o), a;
    }
    function We(t, r) {
      X.set(t, r);
    }
    function Ie(t, r, n, e) {
      var a = R[e + 40 >> 2], o = { tm_sec: R[e >> 2], tm_min: R[e + 4 >> 2], tm_hour: R[e + 8 >> 2], tm_mday: R[e + 12 >> 2], tm_mon: R[e + 16 >> 2], tm_year: R[e + 20 >> 2], tm_wday: R[e + 24 >> 2], tm_yday: R[e + 28 >> 2], tm_isdst: R[e + 32 >> 2], tm_gmtoff: R[e + 36 >> 2], tm_zone: a ? Gt(a) : "" }, s = Gt(n), u = { "%c": "%a %b %d %H:%M:%S %Y", "%D": "%m/%d/%y", "%F": "%Y-%m-%d", "%h": "%b", "%r": "%I:%M:%S %p", "%R": "%H:%M", "%T": "%H:%M:%S", "%x": "%m/%d/%y", "%X": "%H:%M:%S", "%Ec": "%c", "%EC": "%C", "%Ex": "%m/%d/%y", "%EX": "%H:%M:%S", "%Ey": "%y", "%EY": "%Y", "%Od": "%d", "%Oe": "%e", "%OH": "%H", "%OI": "%I", "%Om": "%m", "%OM": "%M", "%OS": "%S", "%Ou": "%u", "%OU": "%U", "%OV": "%V", "%Ow": "%w", "%OW": "%W", "%Oy": "%y" };
      for (var c in u)
        s = s.replace(new RegExp(c, "g"), u[c]);
      var l = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"], p = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
      function y(h, d, S) {
        for (var D = typeof h == "number" ? h.toString() : h || ""; D.length < d; )
          D = S[0] + D;
        return D;
      }
      function v(h, d) {
        return y(h, d, "0");
      }
      function $(h, d) {
        function S(ot) {
          return ot < 0 ? -1 : ot > 0 ? 1 : 0;
        }
        var D;
        return (D = S(h.getFullYear() - d.getFullYear())) === 0 && (D = S(h.getMonth() - d.getMonth())) === 0 && (D = S(h.getDate() - d.getDate())), D;
      }
      function A(h) {
        switch (h.getDay()) {
          case 0:
            return new Date(h.getFullYear() - 1, 11, 29);
          case 1:
            return h;
          case 2:
            return new Date(h.getFullYear(), 0, 3);
          case 3:
            return new Date(h.getFullYear(), 0, 2);
          case 4:
            return new Date(h.getFullYear(), 0, 1);
          case 5:
            return new Date(h.getFullYear() - 1, 11, 31);
          case 6:
            return new Date(h.getFullYear() - 1, 11, 30);
        }
      }
      function j(h) {
        var d = Re(new Date(h.tm_year + 1900, 0, 1), h.tm_yday), S = new Date(d.getFullYear(), 0, 4), D = new Date(d.getFullYear() + 1, 0, 4), ot = A(S), Rt = A(D);
        return $(ot, d) <= 0 ? $(Rt, d) <= 0 ? d.getFullYear() + 1 : d.getFullYear() : d.getFullYear() - 1;
      }
      var E = { "%a": function(h) {
        return l[h.tm_wday].substring(0, 3);
      }, "%A": function(h) {
        return l[h.tm_wday];
      }, "%b": function(h) {
        return p[h.tm_mon].substring(0, 3);
      }, "%B": function(h) {
        return p[h.tm_mon];
      }, "%C": function(h) {
        var d = h.tm_year + 1900;
        return v(d / 100 | 0, 2);
      }, "%d": function(h) {
        return v(h.tm_mday, 2);
      }, "%e": function(h) {
        return y(h.tm_mday, 2, " ");
      }, "%g": function(h) {
        return j(h).toString().substring(2);
      }, "%G": function(h) {
        return j(h);
      }, "%H": function(h) {
        return v(h.tm_hour, 2);
      }, "%I": function(h) {
        var d = h.tm_hour;
        return d == 0 ? d = 12 : d > 12 && (d -= 12), v(d, 2);
      }, "%j": function(h) {
        return v(h.tm_mday + Me(Ft(h.tm_year + 1900) ? Pr : Ar, h.tm_mon - 1), 3);
      }, "%m": function(h) {
        return v(h.tm_mon + 1, 2);
      }, "%M": function(h) {
        return v(h.tm_min, 2);
      }, "%n": function() {
        return `
`;
      }, "%p": function(h) {
        return h.tm_hour >= 0 && h.tm_hour < 12 ? "AM" : "PM";
      }, "%S": function(h) {
        return v(h.tm_sec, 2);
      }, "%t": function() {
        return "	";
      }, "%u": function(h) {
        return h.tm_wday || 7;
      }, "%U": function(h) {
        var d = h.tm_yday + 7 - h.tm_wday;
        return v(Math.floor(d / 7), 2);
      }, "%V": function(h) {
        var d = Math.floor((h.tm_yday + 7 - (h.tm_wday + 6) % 7) / 7);
        if ((h.tm_wday + 371 - h.tm_yday - 2) % 7 <= 2 && d++, d) {
          if (d == 53) {
            var S = (h.tm_wday + 371 - h.tm_yday) % 7;
            S != 4 && (S != 3 || !Ft(h.tm_year)) && (d = 1);
          }
        } else {
          d = 52;
          var D = (h.tm_wday + 7 - h.tm_yday - 1) % 7;
          (D == 4 || D == 5 && Ft(h.tm_year % 400 - 1)) && d++;
        }
        return v(d, 2);
      }, "%w": function(h) {
        return h.tm_wday;
      }, "%W": function(h) {
        var d = h.tm_yday + 7 - (h.tm_wday + 6) % 7;
        return v(Math.floor(d / 7), 2);
      }, "%y": function(h) {
        return (h.tm_year + 1900).toString().substring(2);
      }, "%Y": function(h) {
        return h.tm_year + 1900;
      }, "%z": function(h) {
        var d = h.tm_gmtoff, S = d >= 0;
        return d = Math.abs(d) / 60, d = d / 60 * 100 + d % 60, (S ? "+" : "-") + ("0000" + d).slice(-4);
      }, "%Z": function(h) {
        return h.tm_zone;
      }, "%%": function() {
        return "%";
      } };
      s = s.replace(/%%/g, "\0\0");
      for (var c in E)
        s.includes(c) && (s = s.replace(new RegExp(c, "g"), E[c](o)));
      s = s.replace(/\0\0/g, "%");
      var W = je(s, !1);
      return W.length > r ? 0 : (We(W, t), W.length - 1);
    }
    function ke(t, r, n, e, a) {
      return Ie(t, r, n, e);
    }
    cr = i.InternalError = kt(Error, "InternalError"), bn(), ct = i.BindingError = kt(Error, "BindingError"), Wn(), Sn(), Yn(), gr = i.UnboundTypeError = kt(Error, "UnboundTypeError"), Kn();
    var He = { q: on, v: un, a: cn, h: fn, l: ln, J: hn, S: pn, n: dn, _: mn, d: sn, ma: gn, V: wn, ca: $n, la: Xn, ka: qn, F: Jn, ba: te, R: ne, T: ee, y: oe, t: ie, Q: ce, L: ye, O: ve, na: ge, da: we, ha: br, ia: be, ja: Ce, K: _e, aa: Te, $: Ee, Y: Oe, Z: De, B: aa, P: pa, E: oa, p: ra, b: Ue, w: ea, fa: ua, c: Ye, i: Je, j: Ke, r: Qe, z: na, x: ta, I: ca, N: fa, D: ia, H: da, W: ya, U: va, k: Le, g: Ve, e: Be, f: ze, M: ha, m: Ze, o: Ne, ga: Xe, u: qe, ea: la, A: sa, s: Ge, G: ma, C: Fe, X: ke };
    an();
    var Z = i._free = function() {
      return (Z = i._free = i.asm.qa).apply(null, arguments);
    }, qt = i._malloc = function() {
      return (qt = i._malloc = i.asm.ra).apply(null, arguments);
    }, Er = function() {
      return (Er = i.asm.ta).apply(null, arguments);
    };
    i.__embind_initialize_bindings = function() {
      return (i.__embind_initialize_bindings = i.asm.ua).apply(null, arguments);
    };
    var g = function() {
      return (g = i.asm.va).apply(null, arguments);
    }, gt = function() {
      return (gt = i.asm.wa).apply(null, arguments);
    }, w = function() {
      return (w = i.asm.xa).apply(null, arguments);
    }, b = function() {
      return (b = i.asm.ya).apply(null, arguments);
    }, Sr = function() {
      return (Sr = i.asm.za).apply(null, arguments);
    }, xr = function() {
      return (xr = i.asm.Aa).apply(null, arguments);
    }, Or = function() {
      return (Or = i.asm.Ba).apply(null, arguments);
    }, Dr = function() {
      return (Dr = i.asm.Ca).apply(null, arguments);
    };
    i.dynCall_viijii = function() {
      return (i.dynCall_viijii = i.asm.Da).apply(null, arguments);
    };
    var Fr = i.dynCall_jiii = function() {
      return (Fr = i.dynCall_jiii = i.asm.Ea).apply(null, arguments);
    }, Mr = i.dynCall_jiiii = function() {
      return (Mr = i.dynCall_jiiii = i.asm.Fa).apply(null, arguments);
    };
    i.dynCall_iiiiij = function() {
      return (i.dynCall_iiiiij = i.asm.Ga).apply(null, arguments);
    }, i.dynCall_iiiiijj = function() {
      return (i.dynCall_iiiiijj = i.asm.Ha).apply(null, arguments);
    }, i.dynCall_iiiiiijj = function() {
      return (i.dynCall_iiiiiijj = i.asm.Ia).apply(null, arguments);
    };
    function Ue(t, r) {
      var n = w();
      try {
        return _(t)(r);
      } catch (e) {
        if (b(n), e !== e + 0)
          throw e;
        g(1, 0);
      }
    }
    function ze(t, r, n, e) {
      var a = w();
      try {
        _(t)(r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function Be(t, r, n) {
      var e = w();
      try {
        _(t)(r, n);
      } catch (a) {
        if (b(e), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function Ve(t, r) {
      var n = w();
      try {
        _(t)(r);
      } catch (e) {
        if (b(n), e !== e + 0)
          throw e;
        g(1, 0);
      }
    }
    function Ye(t, r, n) {
      var e = w();
      try {
        return _(t)(r, n);
      } catch (a) {
        if (b(e), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function Le(t) {
      var r = w();
      try {
        _(t)();
      } catch (n) {
        if (b(r), n !== n + 0)
          throw n;
        g(1, 0);
      }
    }
    function Ne(t, r, n, e, a, o) {
      var s = w();
      try {
        _(t)(r, n, e, a, o);
      } catch (u) {
        if (b(s), u !== u + 0)
          throw u;
        g(1, 0);
      }
    }
    function Ge(t, r, n, e, a, o, s, u, c, l, p) {
      var y = w();
      try {
        _(t)(r, n, e, a, o, s, u, c, l, p);
      } catch (v) {
        if (b(y), v !== v + 0)
          throw v;
        g(1, 0);
      }
    }
    function Xe(t, r, n, e, a, o, s) {
      var u = w();
      try {
        _(t)(r, n, e, a, o, s);
      } catch (c) {
        if (b(u), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function qe(t, r, n, e, a, o, s, u) {
      var c = w();
      try {
        _(t)(r, n, e, a, o, s, u);
      } catch (l) {
        if (b(c), l !== l + 0)
          throw l;
        g(1, 0);
      }
    }
    function Je(t, r, n, e) {
      var a = w();
      try {
        return _(t)(r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function Qe(t, r, n, e, a, o) {
      var s = w();
      try {
        return _(t)(r, n, e, a, o);
      } catch (u) {
        if (b(s), u !== u + 0)
          throw u;
        g(1, 0);
      }
    }
    function Ze(t, r, n, e, a) {
      var o = w();
      try {
        _(t)(r, n, e, a);
      } catch (s) {
        if (b(o), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    function Ke(t, r, n, e, a) {
      var o = w();
      try {
        return _(t)(r, n, e, a);
      } catch (s) {
        if (b(o), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    function ta(t, r, n, e, a, o, s) {
      var u = w();
      try {
        return _(t)(r, n, e, a, o, s);
      } catch (c) {
        if (b(u), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function ra(t) {
      var r = w();
      try {
        return _(t)();
      } catch (n) {
        if (b(r), n !== n + 0)
          throw n;
        g(1, 0);
      }
    }
    function na(t, r, n, e, a, o, s) {
      var u = w();
      try {
        return _(t)(r, n, e, a, o, s);
      } catch (c) {
        if (b(u), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function ea(t, r, n, e) {
      var a = w();
      try {
        return _(t)(r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function aa(t, r, n, e) {
      var a = w();
      try {
        return _(t)(r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function oa(t, r, n, e, a, o) {
      var s = w();
      try {
        return _(t)(r, n, e, a, o);
      } catch (u) {
        if (b(s), u !== u + 0)
          throw u;
        g(1, 0);
      }
    }
    function ia(t, r, n, e, a, o, s, u, c, l) {
      var p = w();
      try {
        return _(t)(r, n, e, a, o, s, u, c, l);
      } catch (y) {
        if (b(p), y !== y + 0)
          throw y;
        g(1, 0);
      }
    }
    function ua(t, r, n) {
      var e = w();
      try {
        return _(t)(r, n);
      } catch (a) {
        if (b(e), a !== a + 0)
          throw a;
        g(1, 0);
      }
    }
    function sa(t, r, n, e, a, o, s, u, c, l) {
      var p = w();
      try {
        _(t)(r, n, e, a, o, s, u, c, l);
      } catch (y) {
        if (b(p), y !== y + 0)
          throw y;
        g(1, 0);
      }
    }
    function ca(t, r, n, e, a, o, s, u) {
      var c = w();
      try {
        return _(t)(r, n, e, a, o, s, u);
      } catch (l) {
        if (b(c), l !== l + 0)
          throw l;
        g(1, 0);
      }
    }
    function fa(t, r, n, e, a, o, s, u, c) {
      var l = w();
      try {
        return _(t)(r, n, e, a, o, s, u, c);
      } catch (p) {
        if (b(l), p !== p + 0)
          throw p;
        g(1, 0);
      }
    }
    function la(t, r, n, e, a, o, s, u, c) {
      var l = w();
      try {
        _(t)(r, n, e, a, o, s, u, c);
      } catch (p) {
        if (b(l), p !== p + 0)
          throw p;
        g(1, 0);
      }
    }
    function ha(t, r, n, e, a, o, s) {
      var u = w();
      try {
        _(t)(r, n, e, a, o, s);
      } catch (c) {
        if (b(u), c !== c + 0)
          throw c;
        g(1, 0);
      }
    }
    function pa(t, r, n, e) {
      var a = w();
      try {
        return _(t)(r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function da(t, r, n, e, a, o, s, u, c, l, p, y) {
      var v = w();
      try {
        return _(t)(r, n, e, a, o, s, u, c, l, p, y);
      } catch ($) {
        if (b(v), $ !== $ + 0)
          throw $;
        g(1, 0);
      }
    }
    function ma(t, r, n, e, a, o, s, u, c, l, p, y, v, $, A, j) {
      var E = w();
      try {
        _(t)(r, n, e, a, o, s, u, c, l, p, y, v, $, A, j);
      } catch (W) {
        if (b(E), W !== W + 0)
          throw W;
        g(1, 0);
      }
    }
    function ya(t, r, n, e) {
      var a = w();
      try {
        return Fr(t, r, n, e);
      } catch (o) {
        if (b(a), o !== o + 0)
          throw o;
        g(1, 0);
      }
    }
    function va(t, r, n, e, a) {
      var o = w();
      try {
        return Mr(t, r, n, e, a);
      } catch (s) {
        if (b(o), s !== s + 0)
          throw s;
        g(1, 0);
      }
    }
    var Mt;
    lt = function t() {
      Mt || Rr(), Mt || (lt = t);
    };
    function Rr() {
      if (rt > 0 || (Gr(), rt > 0))
        return;
      function t() {
        Mt || (Mt = !0, i.calledRun = !0, !bt && (Xr(), T(i), i.onRuntimeInitialized && i.onRuntimeInitialized(), qr()));
      }
      i.setStatus ? (i.setStatus("Running..."), setTimeout(function() {
        setTimeout(function() {
          i.setStatus("");
        }, 1), t();
      }, 1)) : t();
    }
    if (i.preInit)
      for (typeof i.preInit == "function" && (i.preInit = [i.preInit]); i.preInit.length > 0; )
        i.preInit.pop()();
    return Rr(), m.ready;
  };
})();
function ja(f) {
  return Yr(Lr, f);
}
async function Wa(f, {
  tryHarder: m = tt.tryHarder,
  formats: i = tt.formats,
  maxNumberOfSymbols: T = tt.maxNumberOfSymbols
} = tt) {
  return Fa(
    f,
    {
      tryHarder: m,
      formats: i,
      maxNumberOfSymbols: T
    },
    Lr
  );
}
const Ia = [
  "aztec",
  "code_128",
  "code_39",
  "code_93",
  "codabar",
  "data_matrix",
  "ean_13",
  "ean_8",
  "itf",
  "pdf417",
  "qr_code",
  "upc_a",
  "upc_e",
  "unknown"
], Nr = /* @__PURE__ */ new Map([
  ["aztec", "Aztec"],
  ["code_128", "Code128"],
  ["code_39", "Code39"],
  ["code_93", "Code93"],
  ["codabar", "Codabar"],
  ["data_matrix", "DataMatrix"],
  ["ean_13", "EAN-13"],
  ["ean_8", "EAN-8"],
  ["itf", "ITF"],
  ["pdf417", "PDF417"],
  ["qr_code", "QRCode"],
  ["upc_a", "UPC-A"],
  ["upc_e", "UPC-E"]
]);
function ka(f) {
  for (const [m, i] of Nr)
    if (f === i)
      return m;
  return "unknown";
}
var wt;
class Ha {
  constructor(m = {}) {
    Ir(this, wt, void 0);
    ja(), kr(this, wt, (m == null ? void 0 : m.formats) ?? []);
  }
  static async getSupportedFormats() {
    return Ia;
  }
  async detect(m) {
    const i = await xa(m);
    return i.width === 0 || i.height === 0 ? [] : (await Wa(i, {
      tryHarder: !0,
      formats: Wr(this, wt).map(
        (P) => Nr.get(P)
      )
    })).map((P) => {
      const {
        topLeft: { x, y: F },
        topRight: { x: L, y: I },
        bottomLeft: { x: M, y: V },
        bottomRight: { x: N, y: U }
      } = P.position, z = Math.min(x, L, M, N), G = Math.min(F, I, V, U), bt = Math.max(x, L, M, N), jt = Math.max(F, I, V, U);
      return {
        boundingBox: new DOMRectReadOnly(z, G, bt - z, jt - G),
        rawValue: P.text,
        format: ka(P.format),
        cornerPoints: [
          {
            x,
            y: F
          },
          {
            x: L,
            y: I
          },
          {
            x: N,
            y: U
          },
          {
            x: M,
            y: V
          }
        ]
      };
    });
  }
}
wt = new WeakMap();
globalThis.BarcodeDetector ?? (globalThis.BarcodeDetector = Ha);
export {
  Ia as BARCODE_DETECTOR_FORMATS,
  Ha as BarcodeDetector,
  za as setZXingModuleOverrides
};
